<div class="container">

<table style="width: 100%;"><tr>
<td>hack_immunophenoscore</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Hack the Immunophenoscore</h2>

<h3>Description</h3>

<p>Obtain various immune biomarkers scores, which combined together give the
immunophenoscore (<em>Charoentong et al., 2017</em>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">hack_immunophenoscore(expr_data, extract = "ips")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>expr_data</code></td>
<td>
<p>A normalized gene expression matrix (or data frame) with
gene symbols as row names and samples as columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extract</code></td>
<td>
<p>A string controlling which type of biomarker scores you want
to obtain. Possible choices are:
</p>

<ul>
<li> <p><code>"ips"</code> (default), only raw and discrete IPS scores;
</p>
</li>
<li> <p><code>"class"</code>, IPS scores together with the four summary class scores;
</p>
</li>
<li> <p><code>"all"</code>, all possible biomarker scores.
</p>
</li>
</ul>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The immunophenoscore is conceived as a quantification of tumor immunogenicity.
It is obtained by aggregating multiple immune biomarkers scores, which are
grouped into four major classes:
</p>

<ul>
<li> <p><em>MHC molecules</em> (<strong>MHC</strong>), expression of MHC class I, class II, and non-classical molecules;
</p>
</li>
<li> <p><em>Immunomodulators</em> (<strong>CP</strong>), expression of certain co-inhibitory and co-stimulatory molecules;
</p>
</li>
<li> <p><em>Effector cells</em> (<strong>EC</strong>), infiltration of activated CD8+/CD4+ T cells and Tem (effector memory) CD8+/CD4+ cells;
</p>
</li>
<li> <p><em>Suppressor cells</em> (<strong>SC</strong>), infiltration of immunosuppressive cells (Tregs and MDSCs).
</p>
</li>
</ul>
<p>The table below shows in detail the 26 immune biomarkers and cell types grouped
by class together with the number of genes which represent them:</p>

<table>
<tr>
<td style="text-align: left;">
   <strong>Class</strong> | </td>
<td style="text-align: center;"> <strong>Biomarker/cell type</strong> | </td>
<td style="text-align: center;"> <strong>No. genes</strong> </td>
</tr>
<tr>
<td style="text-align: left;">
   MHC </td>
<td style="text-align: center;"> B2M </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   MHC </td>
<td style="text-align: center;"> HLA-A </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   MHC </td>
<td style="text-align: center;"> HLA-B </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   MHC </td>
<td style="text-align: center;"> HLA-C </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   MHC </td>
<td style="text-align: center;"> HLA-DPA1 </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   MHC </td>
<td style="text-align: center;"> HLA-DPB1 </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   MHC </td>
<td style="text-align: center;"> HLA-E </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   MHC </td>
<td style="text-align: center;"> HLA-F </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   MHC </td>
<td style="text-align: center;"> TAP1 </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   MHC </td>
<td style="text-align: center;"> TAP2 </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   CP </td>
<td style="text-align: center;"> CD27 </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   CP </td>
<td style="text-align: center;"> CTLA-4 </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   CP </td>
<td style="text-align: center;"> ICOS </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   CP </td>
<td style="text-align: center;"> IDO1 </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   CP </td>
<td style="text-align: center;"> LAG3 </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   CP </td>
<td style="text-align: center;"> PD1 </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   CP </td>
<td style="text-align: center;"> PD-L1 </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   CP </td>
<td style="text-align: center;"> PD-L2 </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   CP </td>
<td style="text-align: center;"> TIGIT </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   CP </td>
<td style="text-align: center;"> TIM3 </td>
<td style="text-align: center;"> 1 </td>
</tr>
<tr>
<td style="text-align: left;">
   EC </td>
<td style="text-align: center;"> Act CD4 </td>
<td style="text-align: center;"> 24 </td>
</tr>
<tr>
<td style="text-align: left;">
   EC </td>
<td style="text-align: center;"> Act CD8 </td>
<td style="text-align: center;"> 26 </td>
</tr>
<tr>
<td style="text-align: left;">
   EC </td>
<td style="text-align: center;"> Tem CD4 </td>
<td style="text-align: center;"> 27 </td>
</tr>
<tr>
<td style="text-align: left;">
   EC </td>
<td style="text-align: center;"> Tem CD8 </td>
<td style="text-align: center;"> 25 </td>
</tr>
<tr>
<td style="text-align: left;">
   SC </td>
<td style="text-align: center;"> MDSC </td>
<td style="text-align: center;"> 20 </td>
</tr>
<tr>
<td style="text-align: left;">
   SC </td>
<td style="text-align: center;"> Treg </td>
<td style="text-align: center;"> 20 </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<h3>Value</h3>

<p>A tibble with one row for each sample in <code>expr_data</code>, a column <code>sample_id</code>
indicating sample identifiers and a number of additional columns depending
on the choice of <code>extract</code>.
</p>


<h3>Algorithm</h3>

<p>Samplewise gene expression z-scores are obtained for each of 26 immune cell
types and biomarkers. Then, weighted averaged z-scores are computed for each
class and the raw immunophenoscore (<code class="reqn">IPS-raw</code>) results as the sum of the
four class scores. Finally, the immunophenoscore (<code class="reqn">IPS</code>) is given as an
integer value between 0 and 10 in the following way:
</p>

<ul>
<li> <p><code class="reqn">IPS = 0</code>, if <code class="reqn">IPS-raw \le 0</code>;
</p>
</li>
<li> <p><code class="reqn">IPS = [10 * (IPS-raw / 3)]</code>, if <code class="reqn">0 &lt; IPS-raw &lt; 3</code>;
</p>
</li>
<li> <p><code class="reqn">IPS = 10</code>, if <code class="reqn">IPS-raw \ge 3</code>.
</p>
</li>
</ul>
<h3>Source</h3>

<p><a href="https://github.com/icbi-lab/Immunophenogram">github.com/icbi-lab/Immunophenogram</a>
</p>


<h3>References</h3>

<p>Charoentong, P., Finotello, F., Angelova, M., Mayer, C.,
Efremova, M., Rieder, D., Hackl, H., &amp; Trajanoski, Z. (2017). Pan-cancer
Immunogenomic Analyses Reveal Genotype-Immunophenotype Relationships and
Predictors of Response to Checkpoint Blockade. <em>Cell reports</em>, 18(1), 248â€“262.
doi: <a href="https://doi.org/10.1016/j.celrep.2016.12.019">10.1016/j.celrep.2016.12.019</a>.
</p>


<h3>See Also</h3>

<p><code>hack_sig()</code> to compute Immunophenoscore biomarkers in different
ways (e.g. use <code>signatures = "ips"</code> and <code>method = "singscore"</code>).
</p>
<p><code>check_sig()</code> to check if all/most of the Immunophenoscore biomarkers are
present in your expression matrix (use <code>signatures = "ips"</code>).
</p>


<h3>Examples</h3>

<pre><code class="language-R">hack_immunophenoscore(test_expr)
hack_immunophenoscore(test_expr, extract = "class")
</code></pre>


</div>