<div class="container">

<table style="width: 100%;"><tr>
<td>sim.seqs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate DNA sequences according to DNA substitution models</h2>

<h3>Description</h3>

<p>Simulates DNA sequences according to various DNA substitution 
models:
</p>

<ul>
<li>
<p> Jukes-Cantor (1969)
</p>
</li>
<li>
<p> Kimura (1980)
</p>
</li>
<li>
<p> Felsenstein (1981)
</p>
</li>
<li>
<p> Hasegawa-Kishino-Yano (1985)
</p>
</li>
</ul>
<p>Output can then be passed to <code>HACReal()</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">sim.seqs(num.seqs, num.haps, length.seqs, count.haps, nucl.freqs, 
codon.tbl = c("standard", "vertebrate mitochondrial", 
"invertebrate mitochondrial"), subst.model = c("JC69", "K80", "F81", "HKY85"), 
mu.rate, transi.rate, transv.rate)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>num.seqs</code></td>
<td>
<p>Number of simulated DNA sequences</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.haps</code></td>
<td>
<p>Number of simulated unique species' haplotypes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>length.seqs</code></td>
<td>
<p>Basepair length of DNA sequences</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>count.haps</code></td>
<td>
<p>Haplotype frequency distribution vector</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nucl.freqs</code></td>
<td>
<p>Nucleotide frequency distribution vector of A, C, G, and 
T respectively</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>codon.tbl</code></td>
<td>
<p>Codon table</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subst.model</code></td>
<td>
<p>Model of DNA substitution</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu.rate</code></td>
<td>
<p>Overall nucleotide mutation rate/site/generation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transi.rate</code></td>
<td>
<p>Nucleotide transition rate/site/generation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transv.rate</code></td>
<td>
<p>Nucleotide transversion rate/site/generation</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A FASTA file of DNA sequences</p>


<h3>Note</h3>

<p><code>num.seqs</code> must be greater than or equal to <code>num.haps</code>.
</p>
<p>Both <code>num.seqs</code> and <code>num.haps</code> must be greater than 1.
</p>
<p><code>nucl.freqs</code> must have a length of four and its elements must sum to 1.
</p>
<p><code>count.haps</code> must have a length of <code>num.haps</code> and its elements must 
sum to <code>num.seqs</code>. 
</p>
<p><code>subst.model</code> must be one of "JC69" (Jukes Cantor corrected p-distance), 
"K80" (Kimura-2-Parameter (K2P), "F81" (Felenstein) or "HKY85"
(Hasegawa-Kishino-Yano)
</p>
<p><code>mu.rate</code> must be specified for both "JC69" and "F81" models
</p>
<p><code>transi.rate</code> and <code>transv.rate</code> must be specified for both "K80" 
and "HKY85" models
</p>
<p>All elements <code>nucl.freqs</code> must be equal to 0.25 when <code>subst.model</code> 
is either "JC69" or "K80" 
</p>
<p>All elements <code>nucl.freqs</code> must differ from 0.25 when <code>subst.model</code> 
is either "F81" or "HKY85"
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Not run: 

# Simulate DNA sequences from the 5'-COI DNA barcode region under a Jukes 
# Cantor nucleotide substitution model

num.seqs &lt;- 100 # number of DNA sequences
num.haps &lt;- 10 # number of haplotypes
length.seqs &lt;- 658 # length of DNA sequences
count.haps &lt;- c(65, rep(10, 2), rep(5, 2), rep(1, 5)) # haplotype frequency distribution
nucl.freqs &lt;- rep(0.25, 4) # nucleotide frequency distribution
subst.model &lt;- "JC69" # desired nucleotide substitution model
codon.tbl &lt;- "vertebrate mitochondrial"
mu.rate &lt;- 1e-3 # mutation rate
transi.rate &lt;- NULL # transition rate
transv.rate &lt;- NULL # transversion rate

sim.seqs(num.seqs = num.seqs, num.haps = num.haps, length.seqs = length.seqs, 
count.haps = count.haps, nucl.freqs = nucl.freqs, codon.tbl = codon.tbl, 
subst.model = subst.model, mu.rate = mu.rate, transi.rate = transi.rate, 
transv.rate = transv.rate)


## End(Not run)

</code></pre>


</div>