<div class="container">

<table style="width: 100%;"><tr>
<td>HACSim-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Iterative Extrapolation of Species' Haplotype Accumulation
Curves for Genetic Diversity Assessment
</h2>

<h3>Description</h3>

<p>HACSim (<strong>H</strong>aplotype <strong>A</strong>ccumulation <strong>C</strong>urve 
<strong>Sim</strong>ulator) employs a novel nonparametric stochastic (Monte Carlo) optimization method of iteratively generating species' haplotype accumulation 
curves through extrapolation to assess sampling completeness based on the 
approach outlined in Phillips et al. (2015) &lt;doi:10.1515/dna-2015-0008&gt;, 
Phillips et al. (2019) &lt;doi:10.1002/ece3.4757&gt; and Phillips et al. (2020) 
&lt;doi: 10.7717/peerj-cs.243&gt;. HACSim outputs a number of useful summary 
statistics of sampling coverage ("Measures of Sampling Closeness"), including 
an estimate of the likely required sample size (along with desired level 
confidence intervals) necessary to recover a given number/proportion of 
observed unique species' haplotypes. Any genomic marker can be targeted to 
assess likely required specimen sample sizes for genetic diversity assessment. 
The method is particularly well-suited to assess sampling sufficiency for DNA barcoding initiatives. Users can also simulate their own DNA sequences 
according to various models of nucleotide substitution. A Shiny app is also 
available.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> HACSim</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Title: </td>
<td style="text-align: left;"> Iterative Extrapolation of Species' Haplotype Accumulation
Curves for Genetic Diversity Assessment</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 1.0.6-1</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2022-05-23</td>
</tr>
<tr>
<td style="text-align: left;">
Author: </td>
<td style="text-align: left;"> Jarrett D. Phillips [aut, cre], 
        Steven H. French [ctb], Navdeep Singh [ctb]</td>
</tr>
<tr>
<td style="text-align: left;">
Maintainer: </td>
<td style="text-align: left;"> Jarrett D. Phillips &lt;phillipsjarrett1@gmail.com&gt;</td>
</tr>
<tr>
<td style="text-align: left;">
Description: </td>
<td style="text-align: left;"> Performs iterative extrapolation of species' haplotype accumulation curves using a nonparametric stochastic (Monte Carlo) optimization method for assessment of specimen sampling completeness based on the approach of Phillips et al. (2015) &lt;doi:10.1515/dna-2015-0008&gt;, Phillips et al. (2019) &lt;doi:10.1002/ece3.4757&gt; and Phillips et al. (2020) &lt;doi: 10.7717/peerj-cs.243&gt;. 'HACSim' outputs a number of useful summary statistics of sampling coverage ("Measures of Sampling Closeness"), including an estimate of the likely required sample size (along with desired level confidence intervals) necessary to recover a given number/proportion of observed unique species' haplotypes. Any genomic marker can be targeted to assess likely required specimen sample sizes for genetic diversity assessment. The method is particularly well-suited to assess sampling sufficiency for DNA barcoding initiatives. Users can also simulate their own DNA sequences according to various models of nucleotide substitution. A Shiny app is also available.</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
<td style="text-align: left;">
URL: </td>
<td style="text-align: left;"> &lt;https://github.com/jphill01/HACSim.R&gt;
&lt;https://github.com/jphill01/HACSim-RShiny-App&gt;
&lt;https://jphill01.shinyapps.io/HACSim&gt;</td>
</tr>
<tr>
<td style="text-align: left;">
NeedsCompilation: </td>
<td style="text-align: left;"> yes</td>
</tr>
<tr>
<td style="text-align: left;">
Imports: </td>
<td style="text-align: left;"> ape (&gt;= 5.3),
data.table (&gt;= 1.12.8),
graphics (&gt;= 3.6.1),
matrixStats (&gt;= 0.56.0),
pegas (&gt;= 0.13),
Rcpp (&gt;= 1.0.3),
shiny (&gt;= 1.6.0),
stats (&gt;= 3.6.1),
utils (&gt;= 3.6.1)</td>
</tr>
<tr>
<td style="text-align: left;">
LinkingTo: </td>
<td style="text-align: left;"> Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td style="text-align: left;">
RoxygenNote: </td>
<td style="text-align: left;"> 6.1.1</td>
</tr>
<tr>
<td style="text-align: left;">
Packaged: </td>
<td style="text-align: left;"> 2019-10-23 16:00:58 UTC; jarrettphillips</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>Index of help topics:
</p>
<pre>
HAC.sim                 Internal R code
HAC.simrep              Run a simulation of haplotype accumulation
                        curves for hypothetical or real species
HACClass                Internal R code
HACHypothetical         Set up an object to simulate haplotype
                        accumulation curves for a hypothetical species
HACReal                 Set up an object to simulate haplotype
                        accumulation curves for a real species
HACSim-package          Iterative Extrapolation of Species' Haplotype
                        Accumulation Curves for Genetic Diversity
                        Assessment
accumulate              Internal C++ code
envr                    Simulation variable storage environment
launchApp               Launch HACSim R Shiny web app
sim.seqs                Simulate DNA sequences according to DNA
                        substitution models
</pre>


<h3>Author(s)</h3>

<p>Jarrett D. Phillips [aut, cre], 
        Steven H. French [ctb], Navdeep Singh [ctb]
</p>
<p>Maintainer: Jarrett D. Phillips &lt;phillipsjarrett1@gmail.com&gt;
</p>


<h3>References</h3>

<p>Phillips, J.D., Gwiazdowski, R.A., Ashlock, D. and Hanner, R. (2015). An 
exploration of sufficient sampling effort to describe intraspecific DNA barcode haplotype diversity: examples from the ray-finned fishes (Chordata: 
Actinopterygii). DNA Barcodes, 3: 66-73.
</p>
<p>Phillips, J.D., Gillis, D.J. and Hanner, R.H. (2019). Incomplete estimates of 
genetic diversity within species: Implications for DNA barcoding. Ecology and Evolution, 9(5): 2996-3010.
</p>
<p>Phillips, J.D., Gillis, D.J. and Hanner, R.H. (2020). HACSim: An R package to estimate intraspecific sample sizes for genetic diversity assessment using 
haplotype accumulation curves. PeerJ Computer Science
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Simulate hypothetical species ##

N &lt;- 100 # total number of sampled individuals
Hstar &lt;- 10 # total number of haplotypes
probs &lt;- rep(1/Hstar, Hstar) # equal haplotype frequency distribution

HACSObj &lt;- HACHypothetical(N = N, Hstar = Hstar, 
probs = probs, filename = "output") # outputs a CSV 
# file called "output.csv"

## Simulate hypothetical species - subsampling ##
HACSObj &lt;- HACHypothetical(N = N, Hstar = Hstar, 
probs = probs, perms = 1000, p = 0.95, 
subsample = TRUE, prop = 0.25, conf.level = 0.95, 
filename = "output")

## Simulate hypothetical species and all paramaters changed - subsampling ##
HACSObj &lt;- HACHypothetical(N = N, Hstar = Hstar, probs = probs, 
perms = 10000, p = 0.90, subsample = TRUE, prop = 0.15, 
conf.level = 0.95, filename = "output")

HAC.simrep(HACSObj) # runs a simulation


## Simulate real species ##

## Not run: 
## Simulate real species ##
# outputs file called "output.csv"
HACSObj &lt;- HACReal(filename = "output") 

## Simulate real species - subsampling ##
HACSObj &lt;- HACReal(subsample = TRUE, prop = 0.15, 
conf.level = 0.95, filename = "output")

## Simulate real species and all parameters changed - subsampling ##
HACSObj &lt;- HACReal(perms = 10000, p = 0.90, subsample = TRUE, 
prop = 0.15, conf.level = 0.99, filename = "output")

# user prompted to select appropriate FASTA file
HAC.simrep(HACSObj) 
## End(Not run)

## Not run: 
## Simulate DNA sequences ##

num.seqs &lt;- 100 # number of DNA sequences
num.haps &lt;- 15 # number of haplotypes
length.seqs &lt;- 658 # length of DNA sequences
count.haps &lt;- c(60, rep(10, 2), rep(5, 2), rep(1, 5)) # haplotype frequency distribution
nucl.freqs &lt;- rep(0.25, 4) # nucleotide frequency distribution
subst.model &lt;- "JC69" # desired nucleotide substitution model
mu.rate &lt;- 1e-3 # mutation rate
transi.rate &lt;- NULL # transition rate
transv.rate &lt;- NULL # transversion rate

sim.seqs(num.seqs = num.seqs, num.haps = num.haps, length.seqs = length.seqs,
nucl.freqs = nucl.freqs, count.haps = count.haps, subst.model = subst.model, 
transi.rate = transi.rate, transv.rate = transv.rate)

# outputs file called "output.csv"
HACSObj &lt;- HACReal(filename = "output") 

## Simulate DNA sequences - subsampling ##
HACSObj &lt;- HACReal(subsample = TRUE, prop = 0.15, 
conf.level = 0.95, filename = "output")

## Simulate DNA sequences and all parameters changed - subsampling ##
HACSObj &lt;- HACReal(perms = 10000, p = 0.90, subsample = TRUE, 
prop = 0.15, conf.level = 0.99, filename = "output") 

# user prompted to select appropriate FASTA file
HAC.simrep(HACSObj) 
## End(Not run)

</code></pre>


</div>