<div class="container">

<table style="width: 100%;"><tr>
<td>HACReal</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Set up an object to simulate haplotype accumulation curves for a real 
species</h2>

<h3>Description</h3>

<p>Helper function which creates an object containing necessary 
information to run a simulation of haplotype accumulation for a real species of interest</p>


<h3>Usage</h3>

<pre><code class="language-R">HACReal(perms = 10000, p = 0.95, conf.level = 0.95, 
ci.type = "quantile", subsample = FALSE, prop = NULL, 
progress = TRUE, num.iters = NULL, filename = NULL)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>perms</code></td>
<td>
<p>Number of permutations (replications)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>Proportion of haplotypes to recover</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>conf.level</code></td>
<td>
<p>Desired confidence level for graphical output and interval estimation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ci.type</code></td>
<td>
<p>Type of confidence interval for graphical output. Choose from "quantile" or "asymptotic"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subsample</code></td>
<td>
<p>Is a subsample of DNA sequences desired?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prop</code></td>
<td>
<p>If subsample = TRUE, the proportion of DNA sequences to subsample</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.iters</code></td>
<td>
<p>Number of iterations to compute</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>progress</code></td>
<td>
<p>Should iteration output be printed to the R console?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>Name of file where simulation results are to be saved</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list object of class "HAC" with 13 elements that can be passed to 
<code>HAC.simrep</code> as follows:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>input.seqs</code></td>
<td>
<p>Should a FASTA file of aligned/trimmed DNA sequences be 
inputted? Default is TRUE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subset.seqs</code></td>
<td>
<p>Should a subsample of DNA sequences be taken? Default is 
FALSE (can be altered by user)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prop.seqs</code></td>
<td>
<p>Proportion of DNA sequences to subsample. Default is NA 
(can be altered by user)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prop.haps</code></td>
<td>
<p>Proportion of haplotype labels to subsample. Default is 
NULL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subset.haps</code></td>
<td>
<p>Should a subsample of haplotype labels be taken? Default 
is NULL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N</code></td>
<td>
<p>Number of individuals. NA by default (computed automatically by 
algorithm)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Hstar</code></td>
<td>
<p>Number of unique species' haplotypes. NA by default (computed automatically by algorithm)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>probs</code></td>
<td>
<p>Haplotype frequency distribution vector. NA by default (computed automatically by algorithm)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>Proportion of haplotypes to recover. <code>p</code> = 0.95 by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perms</code></td>
<td>
<p>Number of permutations (replications). <code>perms</code> = 10000 by 
default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>conf.level</code></td>
<td>
<p>Desired confidence level for graphical output and interval estimation. <code>conf.level</code> = 0.95 by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ci.type</code></td>
<td>
<p>Type of confidence interval for graphical output. 
<code>ci.type = "quantile"</code> by default</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.iters</code></td>
<td>
<p>Number of iterations to compute. <code>num.iters = NULL</code> by 
default (i.e., all iterations are computed; users can specify 
<code>num.iters</code> = 1 for the first iteration.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>progress</code></td>
<td>
<p>Should iteration output be printed to the R console? Default 
is TRUE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>Name of file where simulation results are to be saved.</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">    ## Simulate real species ##
    # outputs file called "output.csv"
    HACSObj &lt;- HACReal(filename = "output") 
    
    ## Simulate real species - subsampling ##
    # subsamples 25% of DNA sequences
    HACSObj &lt;- HACReal(subsample = TRUE, prop = 0.25, conf.level = 0.95, 
    filename = "output") 
    
    ## Simulate real species and all parameters changed - subsampling ##
    HACSObj &lt;- HACReal(perms = 10000, p = 0.90, subsample = TRUE, 
    prop = 0.15, conf.level = 0.99, num.iters = 1, filename = "output")
</code></pre>


</div>