<div class="container">

<table style="width: 100%;"><tr>
<td>run_joint_hmm_s15</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Run 15-state joint HMM on a pseudobulk profile</h2>

<h3>Description</h3>

<p>Run 15-state joint HMM on a pseudobulk profile
</p>


<h3>Usage</h3>

<pre><code class="language-R">run_joint_hmm_s15(
  pAD,
  DP,
  p_s,
  Y_obs = 0,
  lambda_ref = 0,
  d_total = 0,
  theta_min = 0.08,
  theta_neu = 0,
  bal_cnv = TRUE,
  phi_del = 2^(-0.25),
  phi_amp = 2^(0.25),
  phi_bamp = phi_amp,
  phi_bdel = phi_del,
  mu = 0,
  sig = 1,
  r = 0.015,
  t = 1e-05,
  gamma = 18,
  prior = NULL,
  exp_only = FALSE,
  allele_only = FALSE,
  classify_allele = FALSE,
  debug = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pAD</code></td>
<td>
<p>integer vector Paternal allele counts</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>DP</code></td>
<td>
<p>integer vector Total alelle counts</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_s</code></td>
<td>
<p>numeric vector Phase switch probabilities</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Y_obs</code></td>
<td>
<p>numeric vector Observed gene counts</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda_ref</code></td>
<td>
<p>numeric vector Reference expression rates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d_total</code></td>
<td>
<p>integer Total library size for expression counts</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theta_min</code></td>
<td>
<p>numeric Minimum haplotype imbalance threshold</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theta_neu</code></td>
<td>
<p>numeric Haplotype imbalance threshold for neutral state</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bal_cnv</code></td>
<td>
<p>logical Whether to include balanced CNV states</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phi_del</code></td>
<td>
<p>numeric Expected fold change for deletion</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phi_amp</code></td>
<td>
<p>numeric Expected fold change for amplification</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phi_bamp</code></td>
<td>
<p>numeric Expected fold change for balanced amplification</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phi_bdel</code></td>
<td>
<p>numeric Expected fold change for balanced deletion</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu</code></td>
<td>
<p>numeric Global expression bias</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sig</code></td>
<td>
<p>numeric Global expression variance</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r</code></td>
<td>
<p>numeric Variant mapping bias</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>t</code></td>
<td>
<p>numeric Transition probability between copy number states</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gamma</code></td>
<td>
<p>numeric Overdispersion in the allele-specific expression</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prior</code></td>
<td>
<p>numeric vector Prior probabilities for each state</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exp_only</code></td>
<td>
<p>logical Whether to only use expression data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele_only</code></td>
<td>
<p>logical Whether to only use allele data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>classify_allele</code></td>
<td>
<p>logical Whether to classify allele states</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>debug</code></td>
<td>
<p>logical Whether to print debug messages</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional parameters</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>character vector Decoded states
</p>


<h3>Examples</h3>

<pre><code class="language-R">with(bulk_example, {
    run_joint_hmm_s15(pAD = pAD, DP = DP, p_s = p_s, Y_obs = Y_obs, lambda_ref = lambda_ref, 
    d_total = na.omit(unique(d_obs)), mu = mu, sig = sig, t = 1e-5, gamma = 30, theta_min = 0.08)
})
</code></pre>


</div>