<div class="container">

<table style="width: 100%;"><tr>
<td>anova.hapassoc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Return likelihood ratio test of haplotype effect</h2>

<h3>Description</h3>

<p>This function returns the likelihood ratio test statistic comparing
two nested models fit with <code>hapassoc</code> for cohort or cross-sectional
data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'hapassoc'
anova(object, redfit, display=TRUE, ...)</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>a list of class <code>hapassoc</code> output by the 
<code>hapassoc</code> function</p>
</td>
</tr></table>
<p>.
</p>
<table>
<tr style="vertical-align: top;">
<td><code>redfit</code></td>
<td>
<p>A <code>hapassoc</code> object resulting from fitting a reduced 
model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>display</code></td>
<td>
<p>An indicator to suppress output displayed on screen</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional arguments to the summary function currently unused</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>See the hapassoc vignette, of the same name as the package, for details.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>LRTstat</code></td>
<td>
<p>The likelihood ratio statistic comparing the two models</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>Degrees of freedom of the likelihood ratio statistic</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvalue</code></td>
<td>
<p>The p-value of the test</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>Burkett K, McNeney B, Graham J (2004).
A note on inference of trait associations with SNP
haplotypes and other attributes in generalized linear models.
Human Heredity, <b>57</b>:200-206
</p>
<p>Burkett K, Graham J and McNeney B (2006). hapassoc: Software for 
Likelihood Inference of Trait Associations with SNP Haplotypes and Other 
Attributes. Journal of Statistical Software, <b>16(2)</b>:1-19
</p>


<h3>See Also</h3>

<p><code>pre.hapassoc</code>,<code>hapassoc</code>, 
<code>summary.hapassoc</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(hypoDatGeno)
example2.pre.hapassoc&lt;-pre.hapassoc(hypoDatGeno, numSNPs=3, allelic=FALSE)
example2.regr &lt;- hapassoc(affected ~ attr + hAAA+ hACA + hACC + hCAA + 
pooled, example2.pre.hapassoc, family=binomial())
example2.regr2 &lt;- hapassoc(affected ~ attr + hAAA, example2.pre.hapassoc, 
family=binomial())
anova(example2.regr,example2.regr2)

# Returns:

#	hapassoc: likelihood ratio test

#Full model: affected ~ attr + hAAA + hACA + hACC + hCAA + pooled 
#Reduced model: affected ~ attr + hAAA 

#LR statistic = 1.5433 , df = 4 , p-value =  0.8189 
</code></pre>


</div>