<div class="container">

<table style="width: 100%;"><tr>
<td>logLik.hapassoc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Return log-likelihood</h2>

<h3>Description</h3>

<p>This function is used to return the log-likelihood at the maximum 
likelihood estimates computed by <code>hapassoc</code> and to return
the number of parameters fit by <code>hapassoc</code> 
(i.e. the degrees of freedom in <code>R</code>) for cohort or
cross-sectional data. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'hapassoc'
logLik(object, ...)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>a list of class <code>hapassoc</code> output by the 
<code>hapassoc</code> function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional arguments to the summary 
function (currently unused)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>See the hapassoc vignette, of the same name as the package, for details.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>logLik</code></td>
<td>
<p>log-likelihood computed at the maximum likelihood estimates
if <code>design="cohort"</code>, or <code>NA</code> if <code>design="cc"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>number of parameters in the model (i.e. regression coefficients,
any dispersion parameters and haplotype 
frequencies). This is not the residual
degrees of freedom, which is the number of subjects minus the number
of parameters estimated.</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>Burkett K, McNeney B, Graham J (2004).
A note on inference of trait associations with SNP
haplotypes and other attributes in generalized linear models.
Human Heredity, <b>57</b>:200-206
</p>
<p>Burkett K, Graham J and McNeney B (2006). hapassoc: Software for
Likelihood Inference of Trait Associations with SNP Haplotypes and Other
Attributes. Journal of Statistical Software, <b>16(2)</b>:1-19
</p>


<h3>See Also</h3>

<p><code>pre.hapassoc</code>,<code>hapassoc</code>,
<code>summary.hapassoc</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(hypoDatGeno)
example2.pre.hapassoc&lt;-pre.hapassoc(hypoDatGeno, numSNPs=3, allelic=FALSE)
example.regr &lt;- hapassoc(affected ~ attr + hAAA+ hACA + hACC + hCAA + 
pooled, example2.pre.hapassoc, family=binomial())
logLik(example.regr)

# Returns:
# Log Lik: -322.1558 (df=14)
</code></pre>


</div>