<div class="container">

<table style="width: 100%;"><tr>
<td>pre.hapassoc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Pre-process the data before fitting it with hapassoc</h2>

<h3>Description</h3>

<p>This function takes as an argument a dataframe with
non-SNP and SNP data and converts the genotype data at single SNPs
(the single-locus genotypes) into haplotype data.
The rows of the input data frame should correspond to
subjects. Single-locus SNP genotypes  may be specified in 
one of two ways: (i) as pairs of columns, with one column for
each allele of the single-locus genotypes (“allelic format”),
or (ii) as columns of two-character genotypes (“genotypic format”).
The SNP data should comprise the
last 2*numSNPs columns (allelic format) or the last numSNPs columns (genotypic
format) of the data frame. 
</p>
<p>If the haplotypes for a subject cannot be
inferred from his or her genotype data, “pseudo-individuals”
representing all possible haplotype combinations consistent with
the single-locus genotypes are considered.
Missing single-locus genotypes, up to a maximum of maxMissingGenos (see
below), are allowed, but subjects with missing data in more than
maxMissingGenos, or with missing non-SNP data, are removed.
Initial estimates of haplotype frequencies are then obtained using the 
EM algorithm applied to the genotype data pooled over all subjects.
Haplotypes with frequencies below a user-specified tolerance (zero.tol)
are assumed not to exist and are removed from further consideration.
(Pseudo-individuals having haplotypes of negligible frequency are deleted and 
the column in the design matrix corresponding to that haplotype is deleted.)
For the remaining haplotypes, those with non-negligible frequency below a 
user-defined pooling tolerance (pooling.tol) are pooled into a single 
category called “pooled” in the design matrix for the risk model. 
However, the frequencies of each of these pooled haplotypes are 
still calculated separately. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">pre.hapassoc(dat,numSNPs,maxMissingGenos=1,pooling.tol = 0.05, 
       zero.tol = 1/(2 * nrow(dat) * 10), allelic=TRUE, verbose=TRUE)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dat</code></td>
<td>
<p>the non-SNP and SNP data as a data frame. The SNP data should comprise the last 2*numSNPs columns (allelic format) or last numSNPs columns (genotypic format).
Missing allelic data should be coded as <code>NA</code> or <code>""</code> and missing genotypic data should be coded as, e.g., <code>"A"</code> if one allele is missing and <code>""</code> if both alleles are missing.  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numSNPs</code></td>
<td>
<p>number of SNPs per haplotype</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxMissingGenos</code></td>
<td>
<p>maximum number of single-locus genotypes with missing data to allow for each subject. (Subjects with more missing data, or with missing non-SNP data are removed.) The default is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pooling.tol</code></td>
<td>
<p>pooling tolerance – by default set to 0.05</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>zero.tol</code></td>
<td>
<p>tolerance for haplotype frequencies below which haplotypes
are assumed not to exist – by default set to 
<code class="reqn">\frac{1}{2*N*10}</code> where N is the number of subjects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allelic</code></td>
<td>
<p>TRUE if single-locus SNP genotypes are in allelic format and FALSE if in genotypic format; default is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>indicates whether or not a list of the genotype
variables used to form haplotypes and a list of other non-genetic
variables should be printed; default is TRUE.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>See the hapassoc vignette, of the same name as the package, for details.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>haplotest</code></td>
<td>
<p>logical, TRUE if some haplotypes had frequency less than <code>zero.tol</code> and are assumed not to exist</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>initFreq</code></td>
<td>
<p>initial estimates of haplotype frequencies</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>zeroFreqHaplos</code></td>
<td>
<p>list of haplotypes assumed not to exist</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pooledHaplos</code></td>
<td>
<p>list of haplotypes pooled into a single category in the design matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>haploDM</code></td>
<td>
<p>Haplotype portion of the data frame <strong>augmented</strong> with pseudo-individuals. Has <code class="reqn">2^{numSNPs}</code> columns scoring number of copies of each haplotype for each pseudo-individual</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nonHaploDM</code></td>
<td>
<p>non-haplotype portion of the data frame <strong>augmented</strong> with pseudo-individuals</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>haploMat</code></td>
<td>
<p>matrix with 2 columns listing haplotype labels for each pseudo-individual</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>wt</code></td>
<td>
<p>vector giving initial weights for each pseudo-individual for 
the EM algorithm</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ID</code></td>
<td>
<p>index for each individual in the original data frame. Note that all pseudo-individuals have the same ID value</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>Burkett K, McNeney B, Graham J (2004). 
A note on inference of trait associations with SNP
haplotypes and other attributes in generalized linear models.
Human Heredity, <b>57</b>:200-206
</p>
<p>Burkett K, Graham J and McNeney B (2006). hapassoc: Software for 
Likelihood Inference of Trait Associations with SNP Haplotypes and 
Other Attributes. Journal of Statistical Software, <b>16(2)</b>:1-19
</p>


<h3>See Also</h3>

<p><code>hapassoc</code>,<code>summary.hapassoc</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#First example data set has single-locus genotypes in "allelic format"
data(hypoDat)
example.pre.hapassoc&lt;-pre.hapassoc(hypoDat, numSNPs=3)

# To get the initial haplotype frequencies:
example.pre.hapassoc$initFreq
#      h000       h001       h010       h011       h100       h101       h110 
#0.25179111 0.26050418 0.23606001 0.09164470 0.10133627 0.02636844 0.01081260 
#      h111 
#0.02148268 
# The '001' haplotype is estimated to be the most frequent

example.pre.hapassoc$pooledHaplos
# "h101" "h110" "h111"
# These haplotypes are to be pooled in the design matrix for the risk model

names(example.pre.hapassoc$haploDM)
# "h000"   "h001"   "h010"   "h011"   "h100"   "pooled"

####
#Second example data set has single-locus genotypes in "genotypic format"
data(hypoDatGeno)
example2.pre.hapassoc&lt;-pre.hapassoc(hypoDatGeno, numSNPs=3, allelic=FALSE)

# To get the initial haplotype frequencies:
example2.pre.hapassoc$initFreq
#      hAAA       hAAC       hACA       hACC       hCAA       hCAC
#0.25179111 0.26050418 0.23606001 0.09164470 0.10133627 0.02636844
#      hCCA           hCCC 
#0.01081260	0.02148268 
# The 'hAAC' haplotype is estimated to be the most frequent

example2.pre.hapassoc$pooledHaplos
#  "hCAC" "hCCA" "hCCC"
# These haplotypes are to be pooled in the design matrix for the risk model

names(example2.pre.hapassoc$haploDM)
#  "hAAA"   "hAAC"   "hACA"   "hACC"   "hCAA"   "pooled"
</code></pre>


</div>