<div class="container">

<table style="width: 100%;"><tr>
<td>hapiAssembleEnd</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Assembly of haplotypes in regions at the end of a chromosome</h2>

<h3>Description</h3>

<p>Assembly of haplotypes in regions at the end of a chromosome
</p>


<h3>Usage</h3>

<pre><code class="language-R">hapiAssembleEnd(gmt, draftHap, consensusHap, k = 300)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gmt</code></td>
<td>
<p>a dataframe of genotype data of gamete cells</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>draftHap</code></td>
<td>
<p>a dataframe with draft haplotype information</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>consensusHap</code></td>
<td>
<p>a dataframe of the consensus haplotype information</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>a numeric value for the number of hetSNPs that will be combined
with markers beyond the framework for assembly. Default is 300</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a dataframe containing phased haplotypes
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class="language-R">finalDraft &lt;- rep(0,500)
names(finalDraft) &lt;- seq_len(500)

ref &lt;- rep(0,500)
alt &lt;- rep(1,500)

gmtDa &lt;- data.frame(gmt1=ref, gmt2=alt, gmt3=ref,
gmt4=ref, gmt5=c(alt[1:250], ref[251:500]),
stringsAsFactors = FALSE)

idx1 &lt;- sort(sample(seq_len(500), 30, replace = FALSE))
idx2 &lt;- sort(sample(seq_len(500), 30, replace = FALSE))
idx3 &lt;- sort(sample(seq_len(500), 30, replace = FALSE))

gmtDa[idx1,1] &lt;- NA
gmtDa[idx2,2] &lt;- NA
gmtDa[idx3,3] &lt;- NA

consensusHap &lt;- data.frame(hap1=rep(0,500),hap2=rep(1,500),
total=rep(5,500),rate=rep(1,500),
confidence=rep('F',500),
stringsAsFactors = FALSE)
rownames(consensusHap) &lt;- seq_len(500)

consensusHap &lt;- hapiAssembleEnd(gmt = gmtDa, draftHap = finalDraft, 
consensusHap = consensusHap, k = 300)
</code></pre>


</div>