<div class="container">

<table style="width: 100%;"><tr>
<td>num2base</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert genotype coded in 0/1 to A/T/C/G</h2>

<h3>Description</h3>

<p>Convert numeric (0/1) coded genotype to base (A/T/C/G) coded
</p>


<h3>Usage</h3>

<pre><code class="language-R">num2base(hap, ref, alt)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>hap</code></td>
<td>
<p>a dataframe of consensus haplotypes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref</code></td>
<td>
<p>a character represents reference allele</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alt</code></td>
<td>
<p>a character represents alternative allele</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a dataframe containing converted haplotypes
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class="language-R">ref &lt;- sample(c('A','T'),500, replace=TRUE)
alt &lt;- sample(c('C','G'),500, replace=TRUE)

consensusHap &lt;- data.frame(hap1=rep(0,500),hap2=rep(1,500),
    total=rep(5,500),rate=rep(1,500),
    confidence=rep('F',500),
    stringsAsFactors = FALSE)
rownames(consensusHap) &lt;- seq_len(500)

hap &lt;- num2base(hap=consensusHap, ref=ref, alt=alt)
</code></pre>


</div>