<div class="container">

<table style="width: 100%;"><tr>
<td>SampleHaplotype</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate a haplotype</h2>

<h3>Description</h3>

<p>Simulates a haplotype, either sampling using population
parameters or through a meiosis event with two parental haplotypes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SampleHaplotype(H0 = NULL,H1 = NULL,genDist,nDec,nChrom = 1,prMut =
1E-5,QTL = F,checkValidity = TRUE) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>H0</code></td>
<td>
<p>If specified, the first parental haplotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>H1</code></td>
<td>
<p>If specified, the second parental haplotype. If neither H0
nor H1 are specified, a new haplotype is sampled from a base
population. Errors message is displayed when only one haplotype is
provided as meiosis occurs between two haplotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genDist</code></td>
<td>
<p>Map size of the simulated genome in Morgan.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nDec</code></td>
<td>
<p>Number of decimals until which marker positions are
rounded. Partially specifies marker density on the chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nChrom</code></td>
<td>
<p>Number of chromosomes, default at 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prMut</code></td>
<td>
<p>Probability of marker bp mutation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>QTL</code></td>
<td>
<p>If TRUE, qtl alleles are inherited to the next
generation. See function <code>AssignQTL</code> for
assigning qtl to a list of haplotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>checkValidity</code></td>
<td>
<p>If TRUE, tests if a pair of haplotypes is
compatible; e.g. if the number of traits in both is equal (or 0) and
if the sizes are equal.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Markers are continually spaced over the whole genome. Marker density
is specified in <code>SampleBaseHaplotype</code>. Position
of 1 alleles is recorded and stored in @snp attribute of the object of
class <code>haplotype</code>.
If QTL is TRUE, haplotypes with QTL's need to be provided. If not,
nothing happens apart from mutations (same prob. as for single
basepairs).
Function <code>SampleHaplotype</code> is generally called by function
<code>SampleHaplotypes</code>.
</p>


<h3>Value</h3>

<p>An object of class <code>haplotype</code>.
</p>


<h3>See Also</h3>

<p><code>SampleHaplotypes</code>,
<code>SampleBaseHaplotype</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">hList &lt;- SampleHaplotypes(nHaplotypes = 20,genDist =
1,nDec = 3,nLoc = 20) ## create objects
h &lt;- SampleHaplotype(H0 = hList[[1]],H1 = hList[[2]],genDist =
1,nDec = 3)
</code></pre>


</div>