<div class="container">

<table style="width: 100%;"><tr>
<td>summary.haplo.em</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Summarize contents of a haplo.em object
</h2>

<h3>Description</h3>

<p>Display haplotype pairs and their posterior probabilities by subject.
Also display a table with number of max haplotype pairs for a
subject versus how many were kept (max vs. used).  
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'haplo.em'
summary(object, show.haplo=FALSE, digits=max(options()$digits-2, 5), nlines=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>

<p>A haplo.em object
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.haplo </code></td>
<td>

<p>Logical. If TRUE, show the alleles of the haplotype pairs, otherwise show
only the recoded values.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>digits</code></td>
<td>

<p>number of significant digits to be printed for numeric values
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nlines</code></td>
<td>

<p>To shorten output, print the first 1:nlines rows of the large data frame.  
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>... </code></td>
<td>

<p>Optional arguments for the summary method
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data.frame with a row for every subject's possible haplotype pairs
and the posterior probabilities of that pair given their genotypes.
</p>


<h3>See Also</h3>

<p><code>haplo.em</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(hla.demo)
geno &lt;- hla.demo[,c(17,18,21:24)]
label &lt;-c("DQB","DRB","B")
keep &lt;- !apply(is.na(geno) | geno==0, 1, any)

save.em.keep &lt;- haplo.em(geno=geno[keep,], locus.label=label)
save.df &lt;- summary(save.em.keep)
save.df[1:10,]
</code></pre>


</div>