<div class="container">

<table style="width: 100%;"><tr>
<td>haplotypes-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Manipulating DNA Sequences and Estimating Unambiguous Haplotype Network with Statistical Parsimony
</h2>

<h3>Description</h3>

<p>This package provides S4 classes and methods for reading and manipulating aligned DNA sequences, supporting an indel coding methods (only simple indel coding method is available in the current version), showing base substitutions and indels, calculating absolute pairwise distances between DNA sequences, and collapses identical DNA sequences into haplotypes or infering haplotypes using user provided absolute pairwise character difference matrix.  This package also includes S4 classes and methods for estimating genealogical relationships among haplotypes using statistical parsimony and plotting parsimony networks. 
</p>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Read example FASTA file.	
f&lt;-system.file("example.fas",package="haplotypes") 
# invalid character 'N' was replaced with '?' with a warning message
x&lt;-read.fas(file=f)
# an object of class 'Dna'
x 

## or load DNA Sequence data set.
data("dna.obj") 
x&lt;-dna.obj
## Not run: 
x

## End(Not run)

## Compute an absolute pairwise character difference matrix from DNA sequences.
# coding gaps using simple indel coding method
d&lt;- distance(x,indels="sic") 
## Not run: 
d

## End(Not run)

## Infer haplotypes using the 'Dna' object.
# coding gaps using simple indel coding method
h&lt;-haplotype(x,indels="s") 
## Not run: 
h

## End(Not run)

## Conduct statistical parsimony analysis with 95% connection limit.
#algortihmic method
## Not run: 
p&lt;-parsimnet(x,prob=.95) 
p
# plot network
plot(p) 

## End(Not run)

## Plotting pie charts on the statistical parsimony network
## Not run: 
data("dna.obj")
x&lt;-dna.obj
h&lt;-haplotypes::haplotype(x)

## Statistical parsimony with 95
p&lt;-parsimnet(x) 

#randomly generated populations
pop&lt;-c("pop1","pop2","pop3","pop4","pop5","pop6","pop7","pop8")
set.seed(5)
pops&lt;-sample(pop,nrow(x),replace=TRUE)

# Plotting with default parameters.
pieplot(p,h,1, pops)


## End(Not run)
</code></pre>


</div>