<div class="container">

<table style="width: 100%;"><tr>
<td>SampleBaseHaplotype</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate a base population haplotype</h2>

<h3>Description</h3>

<p>Simulates a base population haplotype.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SampleBaseHaplotype(genDist,nDec,nLoc,pSnp = seq(0,1,length.out = nLoc))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genDist</code></td>
<td>
<p>Map size of the simulated genome in Morgan.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nDec</code></td>
<td>
<p>Number of decimals until which marker positions are
rounded. Partially specifies marker density on the chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nLoc</code></td>
<td>
<p>Maximal number of snp loci on genome, default is number of
available positions on genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pSnp</code></td>
<td>
<p>Optional,if specified the snp positions. Default:
uniform distribution of loci over the whole genome. Consequence will
be that loci have allele frequency equal to one. If NULL, positions
will be randomly sampled for each haplotype and allele frequencies
will be close to 0, depending on <code class="reqn">nLoc / (genDist*10^{nDec})</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Generally called by function <code>SampleHaplotypes</code>.
</p>


<h3>Value</h3>

<p>An object of class <code>haplotype</code>.
</p>


<h3>See Also</h3>

<p><code>SampleHaplotypes</code>,
<code>SampleBaseHaplotype</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">hList &lt;- sapply(1:10,function(x)SampleBaseHaplotype(genDist = 1,nDec
= 2,nLoc = 50))
</code></pre>


</div>