<div class="container">

<table style="width: 100%;"><tr>
<td>haploList-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Class "haploList"</h2>

<h3>Description</h3>

<p>Class definition of <code>haploList</code>. Extends lists to
contian objects of class <code>haplotype</code>. Attributes are <code>nDec</code>,
the number of decimal positions and <code>genDist</code>, the genome size
measured in Morgan.</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("haploList", ...)</code>.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>.Data</code>:</dt>
<dd>
<p>Object of class <code>"list"</code> containing
objects of class <code>'haplotype'</code></p>
</dd>
<dt>
<code>genDist</code>:</dt>
<dd>
<p>Object of class <code>"numeric"</code>
expressing the genome size in Morgan.</p>
</dd>
<dt>
<code>nDec</code>:</dt>
<dd>
<p>Object of class <code>"integer"</code>
expressing the number of decimal positions of the
haplotypes.</p>
</dd>
<dt>
<code>nChrom</code>:</dt>
<dd>
<p>Object of class <code>"integer"</code>
expressing the number of chromosomes.</p>
</dd>
</dl>
<h3>Extends</h3>

<p>Class <code>"list"</code>, from data part.
Class <code>"vector"</code>, by class "list", distance 2.
</p>


<h3>Methods</h3>


<dl>
<dt>[</dt>
<dd>
<p><code>signature(x = "haploList", i = "ANY", j =
	"missing")</code>: subset and extract from object of class <code>'haploList'</code></p>
</dd>
<dt>c</dt>
<dd>
<p><code>signature(x = "haploList")</code>: concatenate object of
class <code>'haploList'</code>.</p>
</dd>
</dl>
<h3>See Also</h3>

<p><code>SampleHaplotypes</code>,
<code>haplotype</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">showClass("haploList")
</code></pre>


</div>