<div class="container">

<table style="width: 100%;"><tr>
<td>Dna-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Class <code>"Dna"</code> in the Package <span class="pkg">haplotypes</span>
</h2>

<h3>Description</h3>

<p>S4 class to hold DNA sequence data. </p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("Dna", sequence, seqlengths, seqnames)</code>, however reading fasta file using   <code>read.fas</code> function or coerce matrix, data.frame or list objects to a Dna object using <code>as.dna</code> methods is preferable.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>sequence</code>:</dt>
<dd>
<p>Object of class <code>"matrix"</code> containing DNA sequence data, rows represent sequences and columns represent sites. See also ‘Note’.</p>
</dd>
<dt>
<code>seqlengths</code>:</dt>
<dd>
<p>Object of class <code>"numeric"</code> containing the length of each DNA sequence.</p>
</dd>
<dt>
<code>seqnames</code>:</dt>
<dd>
<p>Object of class <code>"character"</code> containing the name of each DNA sequence.</p>
</dd>
</dl>
<h3>Methods</h3>


<dl>
<dt>[</dt>
<dd>
<p><code>signature(x = "Dna", i = "ANY", j = "ANY")</code>: extracts part of a DNA sequence as an object of class matrix.</p>
</dd>
<dt>[&lt;-</dt>
<dd>
<p><code>signature(x = "Dna", i = "ANY", j = "ANY", value = "ANY")</code>: replaces part of a Dna sequence with an object of class <code>"matrix"</code>, <code>"numeric"</code> or <code>"character"</code>.</p>
</dd>
<dt>append</dt>
<dd>
<p><code>signature(x = "Dna", value = "ANY")</code>: combines two Dna objects.</p>
</dd>
<dt>as.data.frame</dt>
<dd>
<p><code>signature(x = "Dna")</code>: coerces an object of class Dna to a data.frame.</p>
</dd>
<dt>as.list</dt>
<dd>
<p><code>signature(x = "Dna")</code>: coerces an object of class Dna to a list; elements of the list are character vectors that contains the DNA sequences of length equal to corresponding value in the slot <code>seqlengths</code>.</p>
</dd>  
<dt>as.matrix</dt>
<dd>
<p><code>signature(x = "Dna")</code>: coerces an object of class Dna to a matrix.</p>
</dd>
<dt>as.numeric</dt>
<dd>
<p><code>signature(x = "Dna")</code>: coerces an object of class Dna to a numeric matrix.</p>
</dd>
<dt>as.DNAbin</dt>
<dd>
<p><code>signature(x = "Dna")</code>: coerces an object of class Dna to a  <code>DNAbin</code> object.</p>
</dd>
<dt>as.phyDat</dt>
<dd>
<p><code>signature(x = "Dna")</code>: coerces an object of class Dna to a  <code>phyDat</code> object.</p>
</dd>
<dt>basecomp</dt>
<dd>
<p><code>signature(x = "Dna")</code>: calculates base composition of Dna object.</p>
</dd>
<dt>boot.dna</dt>
<dd>
<p><code>signature(x = "Dna")</code>: generates single bootstrap replicate.</p>
</dd>
<dt>distance</dt>
<dd>
<p><code>signature(x = "Dna")</code>: computes and returns an absolute pairwise character difference matrix from DNA sequences. </p>
</dd>
<dt>haplotype</dt>
<dd>
<p><code>signature(x = "Dna")</code>: infers haplotypes from DNA sequences. </p>
</dd>
<dt>image</dt>
<dd>
<p><code>signature(x = "Dna")</code>: displays DNA sequences </p>
</dd>
<dt>indelcoder</dt>
<dd>
<p><code>signature(x = "Dna")</code>: supports simple indel coding method. </p>
</dd>
<dt>length</dt>
<dd>
<p><code>signature(x = "Dna")</code>: returns the longest sequence length. </p>
</dd>
<dt>append</dt>
<dd>
<p><code>signature(x = "Dna", value = "ANY")</code>: combines two Dna objects.</p>
</dd>
<dt>names</dt>
<dd>
<p><code>signature(x = "Dna")</code>: gets the names of an object Dna. </p>
</dd>
<dt>names&lt;-</dt>
<dd>
<p><code>signature(x = "Dna")</code>: sets the names of an object Dna. </p>
</dd>
<dt>ncol</dt>
<dd>
<p><code>signature(x = "Dna")</code>: returns the longest sequence length. </p>
</dd>
<dt>nrow</dt>
<dd>
<p><code>signature(x = "Dna")</code>: returns the total sequence number. </p>
</dd>
<dt>pairnei</dt>
<dd>
<p><code>signature(x = "Dna")</code>:  calculates pairwise Nei's average number of differences between populations. </p>
</dd>
<dt>pairPhiST</dt>
<dd>
<p><code>signature(x = "Dna")</code>:  calculates pairwise PhiST between populations. </p>
</dd>
<dt>parsimnet</dt>
<dd>
<p><code>signature(x = "Dna")</code>: estimates genealogies using statistical parsimony.</p>
</dd>
<dt>polymorp</dt>
<dd>
<p><code>signature(x = "Dna")</code>: displays information about DNA polymorphisms of two sequences; indels and base substitutions, respectively.</p>
</dd>
<dt>range</dt>
<dd>
<p><code>signature(object = "Dna")</code>: returns a vector containing the minimum and maximum length of DNA sequences. </p>
</dd>
<dt>remove.gaps</dt>
<dd>
<p><code>signature(object = "Dna")</code>: removes alignment gaps. </p>
</dd>
<dt>rownames</dt>
<dd>
<p><code>signature(object = "Dna")</code>: retrieve the row names of a DNA sequence matrix.</p>
</dd>
<dt>rownames&lt;-</dt>
<dd>
<p><code>signature(object = "Dna")</code>: set the row names of a DNA sequence matrix.</p>
</dd>
<dt>show</dt>
<dd>
<p><code>signature(object = "Dna")</code>: displays Dna object briefly. </p>
</dd>
<dt>subs</dt>
<dd>
<p><code>signature(x = "Dna")</code>: displays information about base substitutions. </p>
</dd>
<dt>tolower</dt>
<dd>
<p><code>signature(x = "Dna")</code>: Translate characters in DNA sequence matrix from upper to lower case.</p>
</dd>
<dt>toupper</dt>
<dd>
<p><code>signature(x = "Dna")</code>: Translate characters in DNA sequence matrix from lower to upper case.</p>
</dd>
<dt>unique</dt>
<dd>
<p><code>signature(x = "Dna")</code>: returns a list with duplicate DNA sequences removed.</p>
</dd>
</dl>
<h3>Note</h3>

<p>Valid characters for the slot <code>sequence</code> are "A","C","G","T","a","c","g","t","-","?". Numeric entries (integers) between 0-5 will be converted to "?","A","C","G","T","-", respectively. Invalid characters will be replaced with "?" with a warning message.
</p>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


</div>