<div class="container">

<table style="width: 100%;"><tr>
<td>pQQ</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>QQ-plot with confidence intervals for a vector of p-values</h2>

<h3>Description</h3>

<p>Produces a QQ-plot of p-values. The x-axis is -log10 of the expected p-values (under a null hypothesis of no effects), the y-axis is -log10 values of the actual p-values. A (pointwise) confidence interval can be drawn, and names of snps/genes corresponding to the most significant ones can be added.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pQQ(pvals, nlabs = 6, conf = 0.95, lim, mark = 0.05, ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pvals</code></td>
<td>
<p>A vector of p-values. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nlabs</code></td>
<td>
<p>The number of (most significant) p-values to be labeled using <code>names(pvals)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>conf</code></td>
<td>
<p>The confidence level of the pointwise confidence band. The default is 0.95. The confidence intervals are computed under the assumption of the p-values being drawn independently from a uniform [0,1] distribution. To leave out the confidence interval, set this to FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lim</code></td>
<td>
<p>A vector of length 2 giving the plot limits (on a log10 scal, for instance c(0,4)). Plot limits are computed automatically. However, if other plot limits are desireable, they can be set using this argument. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mark</code></td>
<td>
<p>By default, the 0.05 significance level is marked by lines. Can be changed to a different value, or set to FALSE. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Other arguments passed on to the plotting function.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The <code>pvals</code> argument should be a vector of p-values to be plotted. If the vector has names corresponding to marker (snp) names, the plot will label some of the most significant points with the marker names.
</p>


<h3>Value</h3>

<p>No value is returned.
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br>
Professor of Biostatistics<br>
Division of Epidemiology<br>
Norwegian Institute of Public Health<br><a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p> Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396.<br><br>
Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


</div>