<div class="container">

<table style="width: 100%;"><tr>
<td>AssignQTL</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Assigns QTL to a list of Haplotypes</h2>

<h3>Description</h3>

<p><code>AssignQTL</code> assigns QTL to the <code>qtl</code> of objects of class
<code>haplotype</code>. <code>ListQTL</code> returns a list of qtl which for the
object of class <code>haploList</code>. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">AssignQTL(hList,nqtl = NA,frqtl = NA,sigma2qtl = NULL,shQTL = 1,scQTL = 1,
                      nTraits = 1,overlap = 0,MAF = 0.1,rmCausSNP =
TRUE)
ListQTL(hList,nqtl = NA,frqtl = NA,sigma2qtl = NULL,shQTL = 1,scQTL = 1,
                      nTraits = 1,overlap = 0,MAF = 0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>hList</code></td>
<td>
<p>List of haplotype objects.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nqtl</code></td>
<td>
<p>If specified, the number of qtl which are placed on the
genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>frqtl</code></td>
<td>
<p>If specified, the fraction of heterozygous SNP loci which
become QTL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma2qtl</code></td>
<td>
<p>If specified, the qtl variance. If
<code>length(sigma2qtl)&lt;nqtl</code>, sigma2qtl is replicated until
obtaining a vector of appropriate length. Can be specified as a list
of length equal to nTraits.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shQTL</code></td>
<td>
<p>If alpha is not specified, shQTL specifies the shape
parameter of the gamma distribution from which allele substitution
effects are sampled.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scQTL</code></td>
<td>
<p>If alpha is not specified, scQTL specifies the scale
parameter of the gamma distribution from which allele substitution
effects are sampled.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nTraits</code></td>
<td>
<p>The number of traits.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>overlap</code></td>
<td>
<p>Numeric between 0 and 1. Specifies the percantage of
pleiotropic QTL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF</code></td>
<td>
<p>Minor Allele Frequency. Loci with maf below MAF are not
considered to become QTL. Do not count for frqtl.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rmCausSNP</code></td>
<td>
<p>Remove causative SNPs. </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of length <code>nHaplotpes</code>.
</p>


<h3>See Also</h3>

<p><code>SampleHaplotype</code> 
</p>


<h3>Examples</h3>

<pre><code class="language-R">hList &lt;- SampleHaplotypes(nHaplotypes = 20,nLoc = 100,genDist =
1,nDec = 3) ## create objects
 hListd &lt;- SampleHaplotypes(orig = hList,genDist = 1,nDec = 3)
hListQTL &lt;- AssignQTL(hList,frqtl = 0.1,MAF = 0.0)
hListd &lt;- SampleHaplotypes(orig = hListQTL,genDist = 1,nDec
= 3,QTL = TRUE)
qtlList &lt;- ListQTL(hList,frqtl = 0.1,MAF = 0.0)
</code></pre>


</div>