<div class="container">

<table style="width: 100%;"><tr>
<td>queryHaploreg</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>This function queries HaploReg web-based tool and returns results.</h2>

<h3>Description</h3>

<p>This function queries HaploReg web-based tool and returns results.
</p>


<h3>Usage</h3>

<pre><code class="language-R">queryHaploreg(
  query = NULL,
  file = NULL,
  study = NULL,
  ldThresh = 0.8,
  ldPop = "EUR",
  epi = "vanilla",
  cons = "siphy",
  genetypes = "gencode",
  url = Haploreg.settings[["base.url"]],
  timeout = 100,
  encoding = "UTF-8",
  querySNP = FALSE,
  fields = NULL,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>query</code></td>
<td>
<p>Query (a vector of rsIDs).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>A text file (one refSNP ID per line).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>study</code></td>
<td>
<p>A particular study. See function <code>getHaploRegStudyList(...)</code>.
Default: <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ldThresh</code></td>
<td>
<p>LD threshold, r2 (select NA to only show query variants). 
Default: 0.8.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ldPop</code></td>
<td>
<p>1000G Phase 1 population for LD calculation. 
Can be: "AFR", "AMR", "ASN". Default: "EUR".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>epi</code></td>
<td>
<p>Source for epigenomes. 
Possible values: <code>vanilla</code> for ChromHMM (Core 15-state model);
<code>imputed</code> for ChromHMM (25-state model using 12 imputed marks);
<code>methyl</code> for H3K4me1/H3K4me3 peaks;
<code>acetyl</code> for H3K27ac/H3K9ac peaks.
Default: <code>vanilla</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cons</code></td>
<td>
<p>Mammalian conservation algorithm. 
Possible values: <code>gerp</code> for GERP,
<code>siphy</code> for SiPhy-omega,
<code>both</code> for both.
Default: <code>siphy</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genetypes</code></td>
<td>
<p>Show position relative to.
Possible values: <code>gencode</code> for Gencode genes;
<code>refseq</code> for RefSeq genes;
<code>both</code> for both.
Default: <code>gencode</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>url</code></td>
<td>
<p>HaploReg url address. 
Default: &lt;https://pubs.broadinstitute.org/mammals/haploreg/haploreg.php&gt;</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>timeout</code></td>
<td>
<p>A <code>timeout</code> parameter for <code>curl</code>.
Default: 100</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>encoding</code></td>
<td>
<p>sets the <code>encoding</code> for correct retrieval web-page content.
Default: <code>UTF-8</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>querySNP</code></td>
<td>
<p>A flag indicating to return query SNPs only. 
Default: <code>FALSE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fields</code></td>
<td>
<p>A set of fields to extract. Refer to the package vignette 
for available fields. Default: <code>All</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Verbosing output. Default: FALSE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame (table) with results similar to 
HaploReg uses.
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(haploR)
data &lt;- queryHaploreg(c("rs10048158","rs4791078"))
head(data)
</code></pre>


</div>