<div class="container">

<table style="width: 100%;"><tr>
<td>getAll</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get alleles</h2>

<h3>Description</h3>

<p>Extract the sequence of snp alleles from a list of objects of
class <code>haplotype</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getAll(hList,what = c("snp","qtl"),removeHomozygotes =
FALSE,translatePos = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>hList</code></td>
<td>
<p>List of objects of class <code>haplotype</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>what</code></td>
<td>
<p>Specifies if snp or qtl alleles are extracted from
haplotype objects.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>removeHomozygotes</code></td>
<td>
<p>If TRUE, homozygote genotypes are
removed from the marker data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>translatePos</code></td>
<td>
<p>Translates marker positions to positions in
Morgan, else keeps the positions as integers.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Function removes homozygous snp loci before extracting these.
</p>


<h3>Value</h3>

<p>Matrix.
</p>


<h3>Examples</h3>

<pre><code class="language-R">example(AssignQTL)
hh &lt;- getAll(hList = hList)
qq &lt;- getAll(hList = hList,what = 'qtl')
</code></pre>


</div>