<div class="container">

<table style="width: 100%;"><tr>
<td>haplo.em.fitter</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Compute engine for haplotype EM algorithm 
</h2>

<h3>Description</h3>

<p>For internal use within the haplo.stats library
</p>


<h3>Usage</h3>

<pre><code class="language-R">haplo.em.fitter(n.loci, n.subject, weight, geno.vec, n.alleles,
                max.haps, max.iter, loci.insert.order, min.posterior, 
                tol, insert.batch.size, random.start, iseed1, iseed2, 
                iseed3, verbose)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n.loci </code></td>
<td>

<p>number of loci in genotype matrix
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.subject </code></td>
<td>

<p>number of subjects in the sample
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weight</code></td>
<td>

<p>numeric weights
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno.vec </code></td>
<td>

<p>vectorized genotype matrix
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.alleles </code></td>
<td>

<p>numeric vector giving number of alleles at each marker
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.haps </code></td>
<td>

<p>maximum unique haplotypes in the sample
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.iter </code></td>
<td>

<p>maximum iterations to perform in the fitter
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loci.insert.order </code></td>
<td>

<p>order to insert loci for progressive insertion
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.posterior </code></td>
<td>

<p>after insertion and maximization, discard haplotype pairs per person
that do not meet minimum posterior prob
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tol</code></td>
<td>

<p>convergence tolerance for E-M steps
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>insert.batch.size </code></td>
<td>

<p>number of markers to insert per batch
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>random.start</code></td>
<td>

<p>logical; if TRUE, allow for random starting values of haplotype frequencies 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iseed1</code></td>
<td>

<p>random seed for algorithm
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iseed2</code></td>
<td>

<p>random seed for algorithm
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iseed3</code></td>
<td>

<p>random seed for algorithm
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>logical, print long, verbose output from E-M steps?
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For internal use within the haplo.stats library
</p>


</div>