<div class="container">

<table style="width: 100%;"><tr>
<td>f.create.snp.names</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Intern function for creating column names for genotype data</h2>

<h3>Description</h3>

<p>Creating column names either from a .map file or generating dummy names
</p>


<h3>Usage</h3>

<pre><code class="language-R">f.create.snp.names(map.file, header = FALSE, ncol, format, design)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>map.file</code></td>
<td>
<p>Filename (with path if the file is not in current directory) of the
.map file holding the SNP names, if available (see Details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>header</code></td>
<td>
<p>Logical: does the map.file contain a header in the first row?
Default: FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncol</code></td>
<td>
<p>Number of columns IN TOTAL in the dataset containing only the genotype data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>Format of data (will influence how data is processed) - choose from:
</p>

<ul>
<li> <p><em>haplin</em> - data already in one row per family,
</p>
</li>
<li> <p><em>ped</em> - data from .ped file, each row represents an individual.
</p>
</li>
</ul>
<p>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>design</code></td>
<td>
<p>The design used in the study - choose from:
</p>

<ul>
<li> <p><em>triad</em> - (default), data includes genotypes of mother, father and child;
</p>
</li>
<li> <p><em>cc</em> - classical case-control;
</p>
</li>
<li> <p><em>cc.triad</em> - hybrid design: triads with cases and controls
</p>
</li>
</ul>
<p>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If the .map file is given, the SNP names are read in, if not dummy names are
created by attaching numbers to "m" prefix. Then, a "_a" and "_b" suffix is
attached to each allele, respectively. Finally, if the design is "triad" or
"cc.triad", the following suffixes are attached: "_m" for the mother's
alleles, "_f" for the father's, and "_c" for the child's.
</p>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li> <p><em>gen.data.colnames</em> - a vector with names of columns, length equal to 
the number of columns in the genotype dataset (i.e., depending on the format and
design).
</p>
</li>
<li> <p><em>marker.names</em> - a vector containing the names of markers, as read in
from 'map.file', or dummy names.
</p>
</li>
</ul>
<h3>Details</h3>

<p>The .map file should contain at least two columns, where the second one contains SNP 
names. Any additional columns should be separated by a whitespace character, but will 
be ignored. The file should contain a header.
</p>


</div>