<div class="container">

<table style="width: 100%;"><tr>
<td>gene_comp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Model Gene Compendium Data Set</h2>

<h3>Description</h3>

<p>A data frame which contains information from 1345 unique wheat lines on the Sst1 solid stem locus.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gene_comp
</code></pre>


<h3>Format</h3>

<p>A data frame with 1345 rows and 7 columns:
</p>

<dl>
<dt>Trait</dt>
<dd>
<p>A short discription of the phenotype associated with the gene</p>
</dd>
<dt>Chromosome</dt>
<dd>
<p>The chromosome where the gene resides</p>
</dd>
<dt>Gene</dt>
<dd>
<p>The name of the gene</p>
</dd>
<dt>Nursery</dt>
<dd>
<p>The program which produced the gene call for the genotype</p>
</dd>
<dt>Line</dt>
<dd>
<p>A breeder assigned line designation</p>
</dd>
<dt>FullSampleName</dt>
<dd>
<p>A designation unique to the line found in the genotypic matrix</p>
</dd>
<dt>Call</dt>
<dd>
<p>A 'call' given for the allelic state. For this package, it is best to format the non desiarable allele as "non_gene" and the heterozygous state as "het_gene".</p>
</dd>
</dl>
<h3>Source</h3>

<p>Generated by Zachary James Winn for the CSU breeding program via USDA-ARS gene reports and in-house gene assays
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("gene_comp") #lazy loads the dataset for use in the package
</code></pre>


</div>