<div class="container">

<table style="width: 100%;"><tr>
<td>get.hapPair</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Get a list of objects for haplotype pairs
</h2>

<h3>Description</h3>

<p>Get a list of objects for modeling haplotype pairs from a set of unique
haplotypes and their frequencies, given the baseline haplotype
</p>


<h3>Usage</h3>

<pre><code class="language-R">get.hapPair(haplo, haplo.freq, base.index)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>haplo</code></td>
<td>

<p>matrix of haplotypes, with rows the different haplotypes and
columns the alleles of the haplotypes. For H haplotypes of L loci,
haplo has dimension H x L.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>haplo.freq </code></td>
<td>

<p>vector of length H for the population haplotype frequencies
(corresponding to the rows of haplo)
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>base.index </code></td>
<td>

<p>integer index of the haplotype considered to be the base-line for
logistic regression (index between 1 and H); often, the most common
haplotype is chosen for the base-line.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>list with components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>p.g</code></td>
<td>

<p>Genotype probability under Hardy-Weinberg Equilibrium, where the
genotype is the haplotype pair
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x.haplo</code></td>
<td>

<p>Design matrix for all pairs of haplotypes, excluding the baseline
haplotype.  Effects are coded to an additive effect for the haplotypes.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>haplo.indx</code></td>
<td>

<p>two-column matrix containing the indices for the haplotypes in x.haplo.
The indices are the row of the haplotype in haplo.
</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">haplo &lt;- rbind(
               c(     1,    2,    2,    1,    2),
               c(     1,    2,    2,    1,    1),
               c(     1,    1,    2,    1,    1),
               c(     1,    2,    1,    1,    2),
               c(     1,    2,    2,    2,    1),
               c(     1,    2,    1,    1,    1),
               c(     1,    1,    2,    2,    1),
               c(     1,    1,    1,    1,    2),
               c(     1,    2,    1,    2,    1),
               c(     1,    1,    1,    2,    1),
               c(     2,    2,    1,    1,    2),
               c(     1,    1,    2,    1,    2),
               c(     1,    1,    2,    2,    2),
               c(     1,    2,    2,    2,    2),
               c(     2,    2,    2,    1,    2),
               c(     1,    1,    1,    1,    1),
               c(     2,    1,    1,    1,    1),
               c(     2,    1,    2,    1,    1),
               c(     2,    2,    1,    1,    1),
               c(     2,    2,    1,    2,    1),
               c(     2,    2,    2,    1,    1))
dimnames(haplo)[[2]] &lt;- paste("loc", 1:ncol(haplo), sep=".")
haplo &lt;- data.frame(haplo)

haplo.freq &lt;- c(0.170020121, 0.162977867, 0.123742455, 0.117706237, 0.097585513, 0.084507042,
                0.045271630, 0.039235412, 0.032193159, 0.019114688, 0.019114688, 0.013078471,
                0.013078471, 0.013078471, 0.013078471, 0.006036217, 0.006036217, 0.006036217,
                0.006036217, 0.006036217, 0.006036217)

hPair &lt;- get.hapPair(haplo, haplo.freq, base.index=1)
names(hPair)
dim(hPair$x.haplo)
</code></pre>


</div>