<div class="container">

<table style="width: 100%;"><tr>
<td>haplotype-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Class "haplotype" ~~~ </h2>

<h3>Description</h3>

<p>Definition of <code>'haplotype'</code> class.</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("haplotype", ...)</code>.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>snp</code>:</dt>
<dd>
<p>Object of class <code>"integer"</code> containing the
positions on the genome where the haplotype has a 1 allele. The
number of snp positions is the genome size in Morgan times the
number of decimal positions, both stored in <code>'haploList'</code> objects. </p>
</dd>
<dt>
<code>qtl</code>:</dt>
<dd>
<p>Object of class <code>"list"</code>. Names of the list
are the QTL positions, on the same scale as snp positions. Objects
in the list are the QTL alleles.</p>
</dd>
<dt>
<code>hID</code>:</dt>
<dd>
<p>Object of class <code>"numeric"</code> identifying the
current object of class <code>'haplotype'</code>.</p>
</dd>
<dt>
<code>phID0</code>:</dt>
<dd>
<p>Object of class <code>"numeric"</code> pointing to
the first parental haplotype.</p>
</dd>
<dt>
<code>phID1</code>:</dt>
<dd>
<p>Object of class <code>"numeric"</code> pointing to
the second parental haplotype.</p>
</dd>
</dl>
<h3>Methods</h3>

<p>No methods defined with class "haplotype" in the signature.
</p>


<h3>See Also</h3>

<p><code>SampleHaplotype</code>,
<code>SampleHaplotypes</code>,
<code>haploList</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">showClass("haplotype") </code></pre>


</div>