<div class="container">

<table style="width: 100%;"><tr>
<td>divlocus</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Diversity score</h2>

<h3>Description</h3>

<p>Compute a measure of genetic diversity at each locus
</p>


<h3>Usage</h3>

<pre><code class="language-R">divlocus(dat)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>dat</code></td>
<td>
<p> A binary matrix, rows are haplotypes and columns are binary markers </p>
</td>
</tr></table>
<h3>Details</h3>

<p>This function implements a measure of diversity for a locus <code class="reqn">j</code>
as in Clayton (2002). If <code class="reqn">z_ij</code> represents the allele <code class="reqn">j</code> of haplotype <code class="reqn">i</code>, for
<code class="reqn">i=1,...,N</code> and assuming that alleles are coded as 0 and 1, the
diversity measure can be written as
</p>
<p style="text-align: center;"><code class="reqn">D_j = 2*N( \sum_{i=1}^N z_{ij}^2 - (\sum_{i=1}^N z_{ij}) ^2 )</code>
</p>


<h3>Value</h3>

<p>A vector containing the diversity measure for all markers
</p>


<h3>Author(s)</h3>

<p>Giovanni Montana</p>


<h3>References</h3>

<p>D. Clayton. Choosing a set of haplotype tagging SNPs from a
larger set of diallelic loci. 2002. <a href="www-gene.cimr.cam.ac.uk/clayton/software/stata/htSNP/htsnp.pdf">www-gene.cimr.cam.ac.uk/clayton/software/stata/htSNP/htsnp.pdf</a></p>


<h3>Examples</h3>

<pre><code class="language-R">data(ACEdata)
divlocus(ACEdata)
</code></pre>


</div>