<div class="container">

<table style="width: 100%;"><tr>
<td>haplosim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Haplotype data simulator</h2>

<h3>Description</h3>

<p>Generates a random sample of haplotypes, given an haplotype object created from a data set</p>


<h3>Usage</h3>

<pre><code class="language-R">haplosim(n, hap, which.snp = NULL, seed = NULL, force.polym = TRUE, summary = TRUE)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>Number of haplotypes to generate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hap</code></td>
<td>
<p>Haplotype object created with <code>haplodata</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>which.snp</code></td>
<td>
<p>A vector specifying which SNPs to include</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>Seed for the random number generator</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>force.polym</code></td>
<td>
<p>if TRUE, all loci are polymorphic</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>summary</code></td>
<td>
<p>if TRUE, additional summary statistics are returned</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list containing:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>Simulated sample</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freqs</code></td>
<td>
<p>Allele frequency vector</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cor</code></td>
<td>
<p>Correlation matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>div</code></td>
<td>
<p>Locus-specific diversity scores</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mse.freqs</code></td>
<td>
<p>MSE of allele frequencies</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mse.cor</code></td>
<td>
<p>MSE of correlations</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p> Giovanni Montana </p>


<h3>References</h3>

<p>Montana, G. HapSim: a simulation tool for generating haplotype data with
pre-specified allele frequencies and LD coefficients. 2005.
</p>


<h3>See Also</h3>

<p>See also <code> haplodata </code></p>


<h3>Examples</h3>

<pre><code class="language-R">
#
# Example 1
#

data(ACEdata)

# create the haplotype object 
x &lt;- haplodata(ACEdata) 

# simulates a first sample of 100 haplotypes using all markers
y1 &lt;- haplosim(100, x) 

# compares allele frequencies in real and simulated samples
plot(x$freqs, y1$freqs, title=paste("MSE:",y1$mse.freqs)); abline(a=0, b=1)

# compares LD coefficients in real and simulated samples
ldplot(mergemats(x$cor, y1$cor), ld.type='r') 

# simulates a second sample of 1000 haplotypes using the first 20 markers only
y2 &lt;- haplosim(1000, which.snp=seq(20), x) 

#
# Example 2
#

# simulate a sample of 500 haplotypes based on the ACE data set
set.seed(100)
data(ACEdata)
n &lt;- 500
x &lt;- haplodata(ACEdata)
y &lt;- haplosim(n, x)

# compute the haplotype frequencies
# an haplotype starts at markers 17 and ends at marker 22
freq1 &lt;- haplofreqs(ACEdata, 17, 22)
freq2 &lt;- haplofreqs(y$data, 17, 22)

# extract the set of haplotypic configurations that are shared
# by real and simulated data and their frequencies
commonhapls &lt;- intersect(names(freq1),names(freq2)) 
cfreq1 &lt;- freq1[commonhapls]
cfreq2 &lt;- freq2[commonhapls]

# compare real vs simulated haplotype frequencies
par(mar=c(10.1, 4.1, 4.1, 2.1), xpd=TRUE)
legend.text &lt;- names(cfreq1)
bp &lt;- barplot(cbind(cfreq1,cfreq2), main="Haplotype Frequencies",
       names.arg=c("Real","Simulated"), col=heat.colors(length(legend.text)))
legend(mean(range(bp)), -0.3, legend.text, xjust = 0.5,
       fill=heat.colors(length(legend.text)), horiz = TRUE)
chisq.test(x=n*cfreq2, p=cfreq1, simulate.p.value = TRUE, rescale.p = TRUE)

</code></pre>


</div>