<div class="container">

<table style="width: 100%;"><tr>
<td>HWPosterior</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Calculation of posterior probabilities and Bayes factors for
Hardy-Weinberg tests at X-chromosomal variants.
</h2>

<h3>Description</h3>

<p>Function <code>HWPosterior</code> calculates posterior probabilities and Bayes
factors for tests for Hardy-Weinberg equilibrium of autosomal and X-chromosomal
variants.
</p>


<h3>Usage</h3>

<pre><code class="language-R">HWPosterior(males, females, verbose = TRUE, prior.af = c(0.5,0.5), prior.gf =
            c(0.333,0.333,0.333), x.linked = FALSE, precision = 0.05) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>males</code></td>
<td>
<p>A vector of male genotype counts. For autosomal variants, this
should be a three-element named vector like (AA,AB,BB); for X-chromosomal
variants it should be a two-element vector giving the counts of the hemizygous
genotypes like (A,B).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>females</code></td>
<td>
<p>A vector of female genotype counts. This should be a three-element
named vector like (AA,AB,BB)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>prints results if <code>verbose = TRUE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prior.af</code></td>
<td>
<p>Beta prior parameters for male and female allele frequencies</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prior.gf</code></td>
<td>
<p>Dirichlet prior parameters for female genotype
frequencies</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x.linked</code></td>
<td>
<p>logical indicating whether the variant is autosomal or
X-chromosomal</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>precision</code></td>
<td>
<p>precision parameter for marginal likelihoods that
require numeric integration</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For X-chromosomal variants, four possible models are considered, and the posterior
probabilities and Bayes factors for each model are calculated.
</p>
<p>For autosomal variants, ten possible scenarios are considered, and the
posterior probabilities for all models are calculated.
</p>
<p>In general, default Dirichlet priors are used for genotype
frequencies, and beta prior are used for allele frequencies.
</p>


<h3>Value</h3>

<p>For X-chromosomal variants, a matrix with posterior probabilities and
Bayes factors will be produced.
For autosomal variants, a vector of posterior probabilities is produced.
</p>


<h3>Author(s)</h3>

<p>Xavi Puig <a href="mailto:xavier.puig@upc.edu">xavier.puig@upc.edu</a> and
Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a>
</p>


<h3>References</h3>

<p>Puig, X., Ginebra, J. and Graffelman, J. (2017) A Bayesian test for Hardy-Weinberg
equilibrium of bi-allelic X-chromosomal markers. Heredity 119(4):226â€“236.
<a href="https://doi.org/10.1038/hdy.2017.30">doi:10.1038/hdy.2017.30</a>.
</p>
<p>Puig, X., Ginebra, J. and Graffelman, J. (2019) Bayesian model selection for the study
of Hardy-Weinberg proportions and homogeneity of gender allele frequencies.
Heredity  123(5), pp. 549-564. <a href="https://doi.org/10.1038/s41437-019-0232-0">doi:10.1038/s41437-019-0232-0</a>
</p>


<h3>See Also</h3>

 <p><code>HWChisq</code>, <code>HWExact</code>, <code>HWExactStats</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">#
# An X-chromosomal example
#
males   &lt;- c(A=43,B=13)
females &lt;- c(AA=26,AB=19,BB=3)
out     &lt;- HWPosterior(males,females,verbose=TRUE,x.linked=TRUE)
#
# An autosomal example
#
data(JPTsnps)
males     &lt;- JPTsnps[1,1:3]
females   &lt;- JPTsnps[1,4:6]
post.prob &lt;- HWPosterior(males,females,x.linked=FALSE)
</code></pre>


</div>