<div class="container">

<table style="width: 100%;"><tr>
<td>HWChisqStats</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Fast computation of chi-square statistics for Hardy-Weinberg equilibrium
</h2>

<h3>Description</h3>

<p><code>HWChisqStats</code> is a function for the fast computation of chi-square
statistics (or the corresponding p-values) for a large set of bi-allelic markers (typically SNPs). 
</p>


<h3>Usage</h3>

<pre><code class="language-R">HWChisqStats(X, x.linked = FALSE, pvalues = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>A matrix with genotype counts, one row per marker. <code>X</code>
should have 5 columns for an X-chromosomal data set and 3 columns
for an autosomal data set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x.linked</code></td>
<td>
<p>Logical indicating whether the markers are autosomal (<code>x.linked=FALSE</code>)
or X-chromosomal (<code>x.linked=TRUE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvalues</code></td>
<td>
<p>Logical indicated whether chi-square statistics should be returned (<code>pvalues=FALSE</code>) or whether p-values should be returned (<code>pvalues=TRUE</code>).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Matrix <code>X</code> should strictly comply with the following format. For
an autosomal dataset is should contain the 3 genotype counts in order
(AA,AB,BB). For an X-chromosomal dataset it should contain the 5
genotype counts in order (A,B,AA,AB,BB) where A and B are the male
counts and AA, AB and BB the female counts.
</p>
<p>This function was written for speed improvement, and should be much
faster than looping over the rows of <code>X</code> with <code>HWChisq</code>. There
is no error checking on the supplied data matrix.
</p>


<h3>Value</h3>

<p>A vector of chi-square statistics</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Graffelman, J. and Weir, B.S. (2016) Testing for Hardy-Weinberg
equilibrium at 2 bi-allelic genetic markers on the X chromosome.
</p>


<h3>See Also</h3>

 <p><code>HWChisq</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">#
# Autosomal example
#
set.seed(123)
X &lt;- HWData(1000,100)
monom &lt;- (X[,2]==0 &amp; X[,1]==0) | (X[,2]==0 &amp; X[,3]==0)
X &lt;- X[!monom,] # exclude monomorphics
Chisq.stats &lt;- HWChisqStats(X,x.linked=FALSE,pvalues=FALSE)
Chisq.pvals &lt;- HWChisqStats(X,x.linked=FALSE,pvalues=TRUE)
#
# X-chromosomal example
#
X &lt;- HWData(1000,100,n.males=50,nA=75,x.linked=TRUE)
Chisq.stats &lt;- HWChisqStats(X,x.linked=TRUE,pvalues=FALSE)
Chisq.pvals &lt;- HWChisqStats(X,x.linked=TRUE,pvalues=TRUE)
</code></pre>


</div>