<div class="container">

<table style="width: 100%;"><tr>
<td>HWTriExact</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Exact test for Hardy-Weinberg equilibrium and equality of allele
frequencies for tri-allelic variants.
</h2>

<h3>Description</h3>

<p>Function <code>HWTriExact</code> does a standard exact test for Hardy-Weinberg
equilibrium of a tri-allelic variant, and also does joint exact tests
for equilibrium and equality of allele frequencies if the genotype
counts are given separately for both sexes
</p>


<h3>Usage</h3>

<pre><code class="language-R">HWTriExact(x, y = NULL, eps = 1e-10, nperm = 17000, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>vector with 6 genotype counts (AA,AB,AC,BB,BC,CC)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>vector with 6 or 3 genotype counts (AA,AB,AC,BB,BC,CC) or (A,B,C)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eps</code></td>
<td>
<p>a tolerance that can be set for comparing exact probabilities</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nperm</code></td>
<td>
<p>number of permutations (only relevant for autosomal
stratified by gender)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>print output or not</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If only <code>x</code> is specified, an exact test for an autosomal variant
with three alleles will be performed.
</p>
<p>If both <code>x</code> and <code>y</code> are supplied as vectors with 6 elements,
a permutation test for HWE and equality of allele frequencies (EAF) for an
autosomal variant is performed, using <code>nperm</code> permutations. The
distribution of the probabilities is returned in <code>pseudodist</code>. The
computational cost of a completed enumeration algorithm can be
prohibitive in this case.
</p>
<p>If <code>x</code> is supplied as a length 6 vector, and <code>y</code> as a length 3
vector, the variant is assumed to be X-chromosomal, <code>x</code> containing
female genotype counts and <code>y</code> containing male genotyep counts. In
this case a joint exact test for HWE and EAF for an X-chromosomal
tri-allelic variant is executed.
</p>
<p>See the examples in the example section below.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pval</code></td>
<td>
<p>The p-value of the sample</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pseudodist</code></td>
<td>
<p>Distribution of probabilities obtained by simulation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pofthesample</code></td>
<td>
<p>The probability of the observed sample</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Graffelman, J. and Weir, B.S. (2017) Multi-allelic exact tests for Hardy-Weinberg equilibrium that account for gender.
Molecular Ecology Resources. 18(3) pp. 461â€“473. <a href="https://doi.org/10.1111/1755-0998.12748">doi:10.1111/1755-0998.12748</a>.
</p>


<h3>See Also</h3>

<p><code>HWPerm.mult</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#
# Autosomal tri-allelic (not accounting for gender)
#

x &lt;- c(AA=20,AB=31,AC=26,BB=15,BC=12,CC=0)
## Not run: out &lt;- HWTriExact(x)

#
# Autosomal tri-allelic accounting for gender
#

males   &lt;- c(A=1,B=21,C=34) 
females &lt;- c(AA=0,AB=1,AC=0,BB=8,BC=24,CC=15)
## Not run: out &lt;- HWTriExact(females,males)

#
# X-chromosomal tri-allelic accounting for gender
#

males   &lt;- c(A=1,B=21,C=34) 
females &lt;- c(AA=0,AB=1,AC=0,BB=8,BC=24,CC=15)
## Not run: out &lt;- HWTriExact(females,males)
</code></pre>


</div>