<div class="container">

<table style="width: 100%;"><tr>
<td>is.mono</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Detects autosomal and X-chromosomal monomorphic variants</h2>

<h3>Description</h3>

<p>Function <code>is.mono</code> tests if a bi-allelic variant is monomorphic or not</p>


<h3>Usage</h3>

<pre><code class="language-R">is.mono(x)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a vector of three or five genotype counts ((AA,AB,BB) or (A,B,AA,AB,BB)), or a
three-column or five-column matrix with genotype counts (variants in rows, columns)</p>
</td>
</tr></table>
<h3>Details</h3>

<p><code>is.mono</code> assumes autosomal variants are coded in three-element vectors or
three-column matrices, whereas X-chromosomal variants are coded in five-element
vectors or five-column matrices.
</p>


<h3>Value</h3>

<p>A logical or vector of logicals</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>Examples</h3>

<pre><code class="language-R">#
# a polymorphic autosomal marker
#
x &lt;- c(AA=10,AB=20,BB=10)
print(is.mono(x))
#
# a monomorphic autosomal marker
#
x &lt;- c(AT=0,AA=100,TT=0)
print(is.mono(x))
#
# an autosomal marker with only heterozygotes
#
x &lt;- c(AT=100,AA=0,TT=0)
print(is.mono(x))
#
# a matrix with low maf autosomal markers
#
set.seed(123)
X &lt;- HWData(100,50,shape1=1,shape2=20)
number.monomorphics &lt;- sum(is.mono(X))
print(number.monomorphics)
#
# a polymorphic X chromosomal marker
#
x &lt;- c(G=24,C=26,GG=12,CC=13,GC=25)
print(is.mono(x))
#
# another polymorphic X chromosomal marker
#
x &lt;- c(G=24,C=1,GG=25,CC=0,GC=0)
is.mono(x)
#
# a monomorphic X chromosomal marker
#
x &lt;- c(G=24,C=0,GG=12,CC=0,GC=0)
is.mono(x)
#
# a matrix with low maf X-chromosomal markers
#
set.seed(123)
Y &lt;- HWData(100,50,shape=1,shape2=20,x.linked = TRUE)
number.monomorphics &lt;- sum(is.mono(Y))
print(number.monomorphics)
</code></pre>


</div>