<div class="container">

<table style="width: 100%;"><tr>
<td>HWMissing</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Test a bi-allelic marker for Hardy-Weinberg equilibrium in the
presence of missing genotype information.
</h2>

<h3>Description</h3>

<p>Function <code>HWMissing</code> imputes missing genotype data with a
multinomial logit model that uses information from allele intensities
and/or neighbouring markers. Multiple imputation algorithms
implemented in the Mice package are used to obtain imputed data sets.
Inference for HWE is carried out by estimating the inbreeding
coefficient or exact p-values for each imputed data set, and
by combining all estimates
using Rubin's pooling rules.
</p>


<h3>Usage</h3>

<pre><code class="language-R">HWMissing(X, imputecolumn = 1, m = 50, coding = c(0,1,2), verbose = FALSE, alpha = 0.05,
    varest = "oneovern", statistic = "chisquare",  alternative =
"two.sided", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>An input data frame. By default, the first column should
contain the SNP with missing values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>imputecolumn</code></td>
<td>
<p>Indicates which column of the supplied data frame
is to be imputed (by default, the first colum, <code>imputecolumn=1</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>The number of imputations (50 by default)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coding</code></td>
<td>
<p>Indicates how the genotype data is coded (e.g. 0 for AA,
1 for AB, and 2 for BB).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
 <p><code>verbose = TRUE</code> prints results, <code>verbose = FALSE</code> is silent.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>significance level (0.05 by default) used when computing
confidence intervals</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>varest</code></td>
<td>
<p>Estimator for the variance of the inbreeding
coefficient. <code>varest="oneovern"</code> is the default and sets the
variance under the null (1/n). <code>varest="bailey"</code> uses an
approximation (see details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>statistic</code></td>
<td>
<p>If <code>statistic = "chisquare"</code> then inbreeding
coefficients (equivalent to chisquare statistics) will be computed
for each imputed data set and then combined. If <code>statistic =
      "exact"</code> then one-sided exact tests will be computed for each
imputed data set and the resulting p-values will be combined.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alternative</code></td>
<td>
<p><code>two.sided</code> (default) will perform a two-sided
test where both an excess and a dearth of heterozygotes count as
evidence against HWE. <code>less</code> is a one-sided test where only
dearth of heterozygotes counts a evidence against HWE,
<code>greater</code> is a one-sided test where only excess of
heterozygotes counts as evidence against HWE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional options for function <code>mice</code> of the Mice package</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function <code>HWMissing</code> tests one genetic marker (e.g. a SNP)
with missings for HWE. By default, this marker is supposed to be the
first column of dataframe <code>X</code>. The other columns of <code>X</code>
contain covariates to be used in the imputation model. Covariates
will typically be other, correlated markers or allele intensities of
the SNP to be imputed. Covariate markers should be coded as factor
variables whereas allele intensities should be numerical
variables. By default, a polytomous regression model will be used to
impute the missings. If the covariates also contain missings, an
imputation method for each column of <code>X</code> can be specified by
using the <code>method</code> of mice (see example below).
</p>
<p>If there are no covariates, missings can be imputed under the MCAR
assumption. In that case, missings are imputed by taking a random
sample from the observed data. This is what <code>HWMissing</code> will do
if no covariates are supplied, <code>X</code> being a single factor
variable.
</p>
<p>Several estimators for the variance of the inbreeding coefficient
have been described in the literature. The asymptotic variance of the
inbreeding coefficient under the null hypothesis is 1/n, and is used
if <code>varest = "oneovern"</code> is used. This is the recommended
option. Alternatively, the approximation described in Weir (p. 66) can be used
with <code>varest = "bailey"</code>.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Res</code></td>
<td>
<p>A vector with the inbreeding coefficient, a confidence interval
for the inbreeding coefficient, a p-value for a HWE test and missing
data statistics.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Xmat</code></td>
<td>
<p>A matrix with the genotypic composition of each of the
<code>m</code> imputed data sets.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Little, R. J. A. and Rubin, D. B. (2002) Statistical analysis with
missing data. Second edition, New York, John Wiley &amp; sons.
</p>
<p>Graffelman, J., S\'anchez, M., Cook, S. and Moreno, V. (2013) Statistical
inference for Hardy-Weinberg proportions in the presence of missing
genotype information. PLoS ONE 8(12):
e83316. <a href="https://doi.org/10.1371/journal.pone.0083316">doi:10.1371/journal.pone.0083316</a>
</p>
<p>Graffelman, J. (2015) Exploring Diallelic Genetic Markers: The HardyWeinberg Package.
<em>Journal of Statistical Software</em> 64(3): 1-23. <a href="https://doi.org/10.18637/jss.v064.i03">doi:10.18637/jss.v064.i03</a>.
</p>


<h3>See Also</h3>

 <p><code>HWChisq</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">data(Markers)
## Not run: 
set.seed(123)
Results &lt;- HWMissing(Markers[,1],m=50,verbose=TRUE)$Res # no covariates, imputation assuming MCAR.
set.seed(123)
Results &lt;- HWMissing(Markers[,1:3],m=50,verbose=TRUE)$Res # impute with two allele intensities.
set.seed(123)
Results &lt;- HWMissing(Markers[,c(1,4,5)],m=50,verbose=TRUE)$Res # impute with two covariate SNPs

## End(Not run)
</code></pre>


</div>