<div class="container">

<table style="width: 100%;"><tr>
<td>HEMDAG-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>HEMDAG: Hierarchical Ensemble Methods for Directed Acyclic Graphs</h2>

<h3>Description</h3>

<p>The HEMDAG package:
</p>

<ul>
<li>
<p> provides an implementation of several Hierarchical Ensemble Methods (HEMs) for Directed Acyclic Graphs (DAGs);
</p>
</li>
<li>
<p> reconciles flat predictions with the topology of the ontology;
</p>
</li>
<li>
<p> can enhance predictions of virtually any flat learning methods by taking into account the hierarchical relationships between ontology classes;
</p>
</li>
<li>
<p> provides biologically meaningful predictions that obey the true-path-rule, the biological and logical rule that governs the internal coherence of biomedical ontologies;
</p>
</li>
<li>
<p> is specifically designed for exploiting the hierarchical relationships of DAG-structured taxonomies, such as the Human Phenotype Ontology (HPO) or the Gene Ontology (GO), but can be safely applied to tree-structured taxonomies as well (as FunCat), since trees are DAGs;
</p>
</li>
<li>
<p> scales nicely both in terms of the complexity of the taxonomy and in the cardinality of the examples;
</p>
</li>
<li>
<p> provides several utility functions to process and analyze graphs;
</p>
</li>
<li>
<p> provides several performance metrics to evaluate HEMs algorithms;
</p>
</li>
</ul>
<p>A comprehensive tutorial showing how to apply HEMDAG to real case bio-medical case studies is available at <a href="https://hemdag.readthedocs.io">https://hemdag.readthedocs.io</a>.
</p>


<h3>Details</h3>

<p>The HEMDAG package implements the following Hierarchical Ensemble Methods for DAGs:
</p>

<ol>
<li> <p><b>HTD-DAG</b>: Hierarchical Top Down (<code>htd</code>);
</p>
</li>
<li> <p><b>GPAV-DAG</b>: Generalized Pool-Adjacent Violators, <cite>Burdakov et al.</cite> (<code>gpav</code>);
</p>
</li>
<li> <p><b>TPR-DAG</b>: True-Path Rule (<code>tpr.dag</code>);
</p>
</li>
<li> <p><b>DESCENS</b>: Descendants Ensemble Classifier (<code>tpr.dag</code>);
</p>
</li>
<li> <p><b>ISO-TPR</b>: Isotonic-True-Path Rule (<code>tpr.dag</code>);
</p>
</li>
<li> <p><b>Max</b>, <b>And</b>, <b>Or</b>: Heuristic Methods, <cite>Obozinski et al.</cite> (<code>obozinski.heuristic.methods</code>);
</p>
</li>
</ol>
<h3>Author(s)</h3>

<p><em>Marco Notaro</em><code class="reqn">^1</code> (<a href="https://orcid.org/0000-0003-4309-2200">https://orcid.org/0000-0003-4309-2200</a>);<br><em>Alessandro Petrini</em><code class="reqn">^1</code> (<a href="https://orcid.org/0000-0002-0587-1484">https://orcid.org/0000-0002-0587-1484</a>);<br><em>Giorgio Valentini</em><code class="reqn">^1</code> (<a href="https://orcid.org/0000-0002-5694-3919">https://orcid.org/0000-0002-5694-3919</a>);<br></p>
<p>Maintainer:
<em>Marco Notaro</em>
<a href="mailto:marco.notaro@unimi.it">marco.notaro@unimi.it</a><br></p>
<p><code class="reqn">^1</code> <a href="https://sites.google.com/site/anacletolaboratory/">AnacletoLab</a>, Computational Biology and Bioinformatics Laboratory, Computer Science Department, University of Milan, Italy</p>


<h3>References</h3>

<p>Marco Notaro, Max Schubach, Peter N. Robinson and Giorgio Valentini,
<em>Prediction of Human Phenotype Ontology terms by means of Hierarchical Ensemble methods</em>,
BMC Bioinformatics 2017, 18(1):449, doi: <a href="https://doi.org/10.1186/s12859-017-1854-y">10.1186/s12859-017-1854-y</a>
</p>


</div>