<div class="container">

<table style="width: 100%;"><tr>
<td>K_arabidopsis</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Marker-based relatedness matrices for 3 populations of Arabidopsis thaliana.
</h2>

<h3>Description</h3>

<p>Marker-based relatedness matrices based on the SNP-data from Horton <em>et al.</em> (2012).
Three matrices are provided:
(a) <code>K_atwell</code>, for the 199 accessions studied in Atwell <em>et al.</em> (2010).
(b) <code>K_hapmap</code>, for a subset of 350 accessions taken from the Arabidopsis hapmap (Li <em>et al.</em>, 2010).
(c) <code>K_swedish</code>, for 304 Swedish accessions.
All of these are part of the world-wide regmap of 1307 accessions, described in Horton <em>et al.</em> (2012).
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(K_atwell); data(K_hapmap); data(K_swedish)</code></pre>


<h3>Format</h3>

<p>Matrices whose row- and column names are the ecotype or seed-stock IDs of the accessions.
</p>


<h3>Details</h3>

<p>The matrices were computed using equation (2.2) in Astle and Balding (2009); see also
Goddard <em>et al.</em> (2009). The <code>heritability</code>-package does not contain functions
to construct relatedness matrices from genotypic data, but such functions can be found in many other software packages.
For example, GCTA (Yang <em>et al.</em>, 2011), LDAK (Speed <em>et al.</em>, 2012), Fast-LMM (Lippert, 2011) and GEMMA (Zhou and Stephens, 2012).
</p>


<h3>References</h3>


<ul>
<li>
<p> W. Astle and D.J. Balding (2009) Population Structure and Cryptic Relatedness in Genetic Association Studies.
<em>Statistical Science</em>, Vol. 24, No. 4, 451-471.
</p>
</li>
<li>
<p> Atwell, S., Y. S. Huang, B. J. Vilhjalmsson, G. Willems, M. Horton, <em>et al.</em> (2010)
Genome-wide association study of 107 phenotypes in Arabidopsis thaliana inbred lines. <em>Nature</em>
465: 627-631.
</p>
</li>
<li>
<p> Goddard, M.E., Naomi R. Wray, Klara Verbyla and Peter M. Visscher (2009) Estimating Effects and Making Predictions from Genome-Wide Marker Data.
<em>Statistical Science</em>, Vol. 24, No. 4, 517-529.
</p>
</li>
<li>
<p> Horton, M. W., A. M. Hancock, Y. S. Huang, C. Toomajian, S. Atwell, <em>et al.</em>
(2012) Genome-wide patterns of genetic variation in worldwide Arabidopsis thaliana accessions
from the RegMap panel. <em>Nature Genetics</em> 44: 212-216.
</p>
</li>
<li>
<p> Li, Y., Y. Huang, J. Bergelson, M. Nordborg, and J. O. Borevitz (2010) Association
mapping of local climate-sensitive quantitative trait loci in arabidopsis thaliana. <em>PNAS</em> vol. 107, number 49.
</p>
</li>
<li>
<p> Lippert, C., J. Listgarten, Y. Liu, C.M. Kadie, R.I. Davidson, <em>et al.</em> (2011) FaST
linear mixed models for genome-wide association studies. <em>Naure methods</em> 8: 833-835.
</p>
</li>
<li>
<p> Speed, D., G. Hemani, M. R. Johnson, and D.J. Balding (2012) Improved heritability
estimation from genome-wide snps. <em>the American journal of human genetics</em> 91: 1011-1021.
</p>
</li>
<li>
<p> Yang, J., S.H. Lee, M.E. Goddard, and P.M. Visscher (2011) GCTA: a tool for genomewide
complex trait analysis. <em>the American journal of human genetics</em> 88: 76-82.
</p>
</li>
<li>
<p> Zhou, X., and M. Stephens, (2012) Genome-wide efficient mixed-model analysis for association
studies. <em>Nature genetics</em> 44: 821-824.
</p>
</li>
</ul>
<h3>See Also</h3>

<p>For phenotypic data for the population described in Atwell <em>et al.</em> (2010), see <code>LD</code> and <code>LDV</code>.
For phenotypic data for the hapmap, see <code>BT_LW_H</code> and <code>LA_H</code>.
For phenotypic data for the Swedish regmap, see <code>LA_S</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(K_atwell)
data(K_hapmap)
data(K_swedish)
</code></pre>


</div>