<div class="container">

<table style="width: 100%;"><tr>
<td>floweringTime</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Flowering time data taken from Atwell <em>et al.</em> (2010).
</h2>

<h3>Description</h3>

<p>Two data-frames containing individual plant data on flowering time under different conditions:
<code>LDV</code> (Flowering time under long days and vernalization) and <code>LD</code> (Flowering time under long days, without vernalization).
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(LD); data(LDV)</code></pre>


<h3>Format</h3>

<p>Data-frames with flowering time observations, genotype and design information:
</p>

<dl>
<dt><code>genotype</code></dt>
<dd>
<p>a factor, the levels being the accession or ecotype identifiers</p>
</dd>
<dt><code>LD</code></dt>
<dd>
<p>Flowering time under long days, in number of days</p>
</dd>
<dt><code>LDV</code></dt>
<dd>
<p>Flowering time under long days and vernalization, in number of days</p>
</dd>
<dt><code>replicate</code></dt>
<dd>
<p>The replicate (or block) each plant is contained in (factor with levels <code>1</code> to <code>6</code>)</p>
</dd>
<dt><code>rep1</code></dt>
<dd>
<p>numeric encoding of the factor replicate: equals 1 if the plant is in replicate 1 and 0 otherwise</p>
</dd>
<dt><code>rep2</code></dt>
<dd>
<p>numeric encoding of the factor replicate: equals 1 if the plant is in replicate 2 and 0 otherwise</p>
</dd>
<dt><code>rep3</code></dt>
<dd>
<p>numeric encoding of the factor replicate: equals 1 if the plant is in replicate 3 and 0 otherwise</p>
</dd>
<dt><code>rep4</code></dt>
<dd>
<p>numeric encoding of the factor replicate: equals 1 if the plant is in replicate 4 and 0 otherwise</p>
</dd>
<dt><code>rep5</code></dt>
<dd>
<p>numeric encoding of the factor replicate: equals 1 if the plant is in replicate 5 and 0 otherwise</p>
</dd>
</dl>
<h3>Details</h3>

<p>All plants that had not flowered by the end of the experiment were given a phenotypic value of 200.
Only accessions for which SNP-data are available are included here: 167 accessions in case of LD and 168 accessions in case of LDV.
</p>


<h3>References</h3>


<ul>
<li>
<p> Atwell, S., Y. S. Huang, B. J. Vilhjalmsson, G. Willems, M. Horton, <em>et al.</em> (2010)
Genome-wide association study of 107 phenotypes in Arabidopsis thaliana inbred lines. Nature
465: 627-631.
</p>
</li>
<li>
<p> Kruijer, W. <em>et al.</em> (2015) Marker-based estimation of heritability in immortal
populations. <em>Genetics</em>, Vol. 199(2), p. 1-20.
</p>
</li>
</ul>
<h3>See Also</h3>

<p>For the corresponding genetic relatedness matrix, see <code>K_atwell</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(LD); data(LDV)
</code></pre>


</div>