<div class="container">

<table style="width: 100%;"><tr>
<td>DE_data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creates a <code>DE_data</code> object</h2>

<h3>Description</h3>

<p>Creates a <code>DE_data</code> object from a tibble or data-frame of differential expression
results.
</p>


<h3>Usage</h3>

<pre><code class="language-R">DE_data(dat, keep_rownames_as = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dat</code></td>
<td>
<p>Tibble, results from a differential expression analysis. See Details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_rownames_as</code></td>
<td>
<p>Character, the name of the column in which to save the
rownames of the input data-frame. Default value is <code>NULL</code>, i.e. rownames will
be discarded.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The input data should have one row per gene or transcript, and at least the
following columns:
</p>

<ul>
<li> <p><code>chromosome</code>: character column, chromosome on which the gene/transcript is located.
</p>
</li>
<li> <p><code>start</code> and <code>end</code>: numeric, starting and end position of the gene/transcript
(in bp). A column <code>position</code> will be constructed as the middle value (mean) between
<code>start</code> and <code>end</code>.
</p>
</li>
<li> <p><code>score</code> or <code>padj</code>: numeric, the DE score or adjusted p-value of the
gene/transcript. If column <code>score</code> column is missing, will be constructed
as <code>-log10(padj)</code>.
</p>
</li>
<li> <p><code>foldChange</code> or <code>log2FoldChange</code>: numeric, the fold-change or log2(fold-change)
of the gene/transcript. If column <code>log2FoldChange</code> is missing, will be constructed
as <code>log2(foldChange)</code>.
</p>
</li>
</ul>
<h3>Value</h3>

<p>A <code>DE_data</code> object, i.e. a tibble.
</p>


<h3>Examples</h3>

<pre><code class="language-R">x &lt;- get_example_data()

DE_data(x[["DE"]])
</code></pre>


</div>