<div class="container">

<table style="width: 100%;"><tr>
<td>biall2dos</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Converts bi-allelic SNPs from hierfstat format to dosage format</h2>

<h3>Description</h3>

<p>Converts bi-allelic SNPs hierfstat format to dosage format, the number of alternate allele copies at a locus
for an individual, i.e. 11 -&gt; 0; 12 or 21 &gt;1 and 22 -&gt;2
</p>


<h3>Usage</h3>

<pre><code class="language-R">biall2dos(dat,diploid=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dat</code></td>
<td>
<p>a hierfstat data frame without the first column (the population identifier), 
individuals in rows, columns with individual genotypes encoded as 11, 12, 21 and 22</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>diploid</code></td>
<td>
<p>whether the data set is from a diploid organism</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a matrix containing allelic dosages
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
biall2dos(sim.genot(nbal=2,nbloc=10)[,-1])  # a 10 column matrix

## End(Not run)

</code></pre>


</div>