<div class="container">

<table style="width: 100%;"><tr>
<td>read.VCF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reads a VCF file into a BED object</h2>

<h3>Description</h3>

<p>Reads a <a href="https://samtools.github.io/hts-specs/">https://samtools.github.io/hts-specs/</a>Variant Call Format (VCF) file into a BED object,
retaining bi-allelic SNPs only
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.VCF(fname,BiAllelic=TRUE,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>fname</code></td>
<td>
<p>VCF file name. The VCF file can be compressed (VCF.gz)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BiAllelic</code></td>
<td>
<p>Logical. If TRUE, only bi-allelic SNPs are retained,
otherwise, all variant are kept</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>other arguments to pass to the function</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A <code>bed.matrix-class</code> object
</p>


<h3>See Also</h3>

<p><code>read.vcf</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"> 
 filepath &lt;-system.file("extdata", "LCT.vcf.gz", package="gaston")
 x1 &lt;- read.VCF( filepath )
 x1

</code></pre>


</div>