<div class="container">

<table style="width: 100%;"><tr>
<td>mlhighHet</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>mlhighHet</h2>

<h3>Description</h3>

<p>This function extracts features based on ML method, finds optimal cut-off values of features using sequencial
Cox PH model and obtain the most consistent level according to the cut-offs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mlhighHet(cols, idSurv, idEvent, idFrail, num, fold = 3, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cols</code></td>
<td>
<p>A numeric vector of column numbers indicating the features for which the log Loss functions are to be computed</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>idSurv</code></td>
<td>
<p>The name of the survival time variable</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>idEvent</code></td>
<td>
<p>The name of the survival event variable</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>idFrail</code></td>
<td>
<p>The name of the frailty variable</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num</code></td>
<td>
<p>Number of features to be selected</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fold</code></td>
<td>
<p>An integer denoting number of folds in cross validation, default value 3</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>A data frame that contains the survival and covariate information for the subjects</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Performs heterogeneity analysis in gene expression
</p>
<p>This function extracts features based on minimum log-Loss function using Cox proportional hazard model as learner method on a high dimensional survival data.
For those selected genes, we obtain optimal cutoff values using minimum p-value in a Cox PH model. The Cox PH model is used sequencially for each combination of genes
and all possible gene combinations are tested to obtain best possible combination with minimum BIC value. The subjects are classified according to different levels of those genes. Using a Cox PH frailty model, we obtain the most consistent level for which the frailty variance is minimum.
The data is splited using cross validation technique. The performance measure is considered as logarithmic loss function. It is defined as,
</p>
<p style="text-align: center;"><code class="reqn">L(f,t)=-log(f(t))</code>
</p>

<p>The CoxPH frailty model is defined as,
</p>
<p style="text-align: center;"><code class="reqn">\lambda(t)=\lambda 0(t)\nu exp{X'\beta}</code>
</p>
<p> where <code class="reqn">\nu</code> is called the frailty. The variance of the
frailty term is considered as the heterogeneity among the subjects or patients. Gaussian distribution with mean 0 is considered for the distribution of frailty component.
</p>


<h3>Value</h3>

<p>dataframes containing optimal gene cutoff values and most consistent level according to those cut-offs with frailty variance.
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee, Gajendra K. Vishwakarma &amp; Souvik Banerjee
</p>


<h3>References</h3>

<p>Sonabend, R., Kir√°ly, F. J., Bender, A., Bernd Bischl B. and Lang M. mlr3proba: An R Package for Machine Learning in Survival Analysis, 2021, Bioinformatics, &lt;https://doi.org/10.1093/bioinformatics/btab039&gt;
</p>
<p>Bhattacharjee, A. Vishwakarma, G.K. and Banerjee, S. A modified risk detection approach of biomarkers by frailty effect on multiple time to event data, 2020, &lt;arXiv:2012.02102&gt;.
</p>


<h3>See Also</h3>

<p>mlhighCox, mlhighFrail
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
data(hnscc)
mlhighHet(cols=c(27:32), idSurv="OS", idEvent="Death", idFrail="ID", num=2, fold = 3, data=hnscc)

## End(Not run)
</code></pre>


</div>