<div class="container">

<table style="width: 100%;"><tr>
<td>hJAM_egger</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Fit hJAM with Egger regression</h2>

<h3>Description</h3>

<p>The hJAM_egger function is to get the results from the hJAM model with Egger regression. It is for detecting potential pleiotropy
</p>


<h3>Usage</h3>

<pre><code class="language-R">hJAM_egger(betas.Gy, N.Gy, Gl, A, ridgeTerm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>betas.Gy</code></td>
<td>
<p>The betas in the paper: the marginal effects of SNPs on the phenotype (Gy)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N.Gy</code></td>
<td>
<p>The sample size of Gy</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Gl</code></td>
<td>
<p>The reference panel (Gl), such as 1000 Genome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>A</code></td>
<td>
<p>The A matrix in the paper: the marginal/conditional effects of SNPs on the exposures (Gx)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ridgeTerm</code></td>
<td>
<p>ridgeTerm = TRUE when the matrix L is singular. Matrix L is obtained from the cholesky decomposition of G0'G0. Default as FALSE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of the hJAM with egger regression results.
</p>

<dl>
<dt>Exposure</dt>
<dd>
<p>The intermediates, such as the modifiable risk factors in Mendelian Randomization and
gene expression in transcriptome analysis.</p>
</dd>
<dt>numSNP</dt>
<dd>
<p>The number of SNPs that the user use in the instrument set.</p>
</dd>
<dt>Estimate</dt>
<dd>
<p>The conditional estimates of the associations between intermediates and the outcome.</p>
</dd>
<dt>StdErr</dt>
<dd>
<p>The standard error of the conditional estimates of the associations between intermediates
and the outcome.</p>
</dd>
<dt>Lower.CI</dt>
<dd>
<p>The lower bound of the 95% confidence interval of the estimates.</p>
</dd>
<dt>Upper.CI</dt>
<dd>
<p>The upper bound of the 95% confidence interval of the estimates.</p>
</dd>
<dt>Pvalue</dt>
<dd>
<p>The p value of the estimates with a type-I error equals 0.05.</p>
</dd>
<dt>Est.Int</dt>
<dd>
<p>The intercept of the regression of intermediates on the outcome.</p>
</dd>
<dt>StdErr.Int</dt>
<dd>
<p>The standard error of the intercept of the regression of intermediates
on the outcome.</p>
</dd>
<dt>Lower.CI.Int</dt>
<dd>
<p>The lower bound of the 95% confidence interval of the intercept.</p>
</dd>
<dt>Upper.CI.Int</dt>
<dd>
<p>The upper bound of the 95% confidence interval of the intercept.</p>
</dd>
<dt>Pvalue.Int</dt>
<dd>
<p>The p value of the intercept with a type-I error equals 0.05.</p>
</dd>
</dl>
<p>An object of hJAM with egger regression results.
</p>


<h3>Author(s)</h3>

<p>Lai Jiang
</p>


<h3>References</h3>

<p>Lai Jiang, Shujing Xu, Nicholas Mancuso, Paul J. Newcombe, David V. Conti (2020).
A Hierarchical Approach Using Marginal Summary Statistics for Multiple Intermediates
in a Mendelian Randomization or Transcriptome Analysis. <em>bioRxiv</em>
<a href="https://doi.org/10.1101/2020.02.03.924241">https://doi.org/10.1101/2020.02.03.924241</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(Gl)
data(betas.Gy)
data(conditional_A)
hJAM_egger(betas.Gy = betas.Gy, Gl = Gl, N.Gy = 459324, A = conditional_A, ridgeTerm = TRUE)
</code></pre>


</div>