<div class="container">

<table style="width: 100%;"><tr>
<td>get_cond_alpha</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute conditional alphas</h2>

<h3>Description</h3>

<p>The get_cond_alpha function is to compute the conditional alpha vector for each X
If only one X in the model, please use get_cond_alpha instead of get_cond_A
A sub-step in the get_cond_A function
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_cond_alpha(alphas, Gl, N.Gx, ridgeTerm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>alphas</code></td>
<td>
<p>the marginal effects of SNPs on one exposure (Gx).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Gl</code></td>
<td>
<p>the reference panel (Gl), such as 1000 Genome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N.Gx</code></td>
<td>
<p>the sample size of the Gx. It can be a scalar.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ridgeTerm</code></td>
<td>
<p>ridgeTerm = TRUE when the matrix L is singular. Matrix L is obtained from the cholesky decomposition of G0'G0. Default as FALSE</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A vector with conditional estimates which are converted from marginal estimates using the JAM model.
</p>


<h3>Author(s)</h3>

<p>Lai Jiang
</p>


<h3>References</h3>

<p>Lai Jiang, Shujing Xu, Nicholas Mancuso, Paul J. Newcombe, David V. Conti (2020).
A Hierarchical Approach Using Marginal Summary Statistics for Multiple Intermediates
in a Mendelian Randomization or Transcriptome Analysis. <em>bioRxiv</em>
<a href="https://doi.org/10.1101/2020.02.03.924241">https://doi.org/10.1101/2020.02.03.924241</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(Gl)
data(betas.Gy)
data(marginal_A)
get_cond_alpha(alphas = marginal_A[, 1], Gl = Gl, N.Gx = 339224, ridgeTerm = TRUE)
</code></pre>


</div>