<div class="container">

<table style="width: 100%;"><tr>
<td>multiSearch</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Search Alignment Sequences at Multiple Positions for a Specified Allele</h2>

<h3>Description</h3>

<p>Generates a character string of multiple protein, codon or nucleotide or genomic sequences at the specified positions for the specified allele name.
</p>


<h3>Usage</h3>

<pre><code class="language-R">multiSearch(
  alignType,
  locus,
  allele,
  positions,
  prefix = TRUE,
  sep = "~",
  trimmed = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>alignType</code></td>
<td>
<p>The type of alignment being searched. Allowed values are "prot", codon", "nuc" and "gen". Only one 'alignType' value is allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locus</code></td>
<td>
<p>A specific locus.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele</code></td>
<td>
<p>An allele name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>positions</code></td>
<td>
<p>Nucleotide positions to search.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prefix</code></td>
<td>
<p>A logical that indicates if the position number(s) should be included in the result.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p>Defines the separator between position sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trimmed</code></td>
<td>
<p>A logical identifying whether 'allele' is a two-field, 'trimmed' allele name (trimmed=TRUE), or a full-length name (trimmed=FALSE). The default value is FALSE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>codons (possibly with position number) at 'positions' for 'allele'. If 'allele' does not have a codon at 'position', ‚Äù will be returned.
</p>


<h3>Note</h3>

<p>For internal HLAtools use only.
</p>


</div>