<div class="container">

<table style="width: 100%;"><tr>
<td>multiAlign</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate an Alignment for Specific Alleles at Different Positions</h2>

<h3>Description</h3>

<p>Generates a peptide, codon, coding nucleotide or genomic alignment for HLA alleles allowing each allele to be aligned to a different set of positions.
</p>


<h3>Usage</h3>

<pre><code class="language-R">multiAlign(alignType, alleles, positions)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>alignType</code></td>
<td>
<p>The type of alignment being searched. Allowed values are "prot", codon", "nuc" and "gen".  Only one 'alignType' value is allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p>A vector of un-prefixed HLA locus names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>positions</code></td>
<td>
<p>A list of vectors of nucleotide positions, exactly one vector for each allele, against which each corresponding allele will be aligned.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame of 'allele' and the corresponding nucleotide sequence for specified positions designated for an allele. a nucleotide sequence is not returned for a specific allele if input allele is not available in the ANHIG/IMGTHLA Github Repository. position will be left empty if specific allele does not have a nucleotide at the input position.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>


</div>