<div class="container">

<table style="width: 100%;"><tr>
<td>customAlign</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate a Customized Peptide, Codon or Nucleotide Sequence Alignment.</h2>

<h3>Description</h3>

<p>Generates a peptide, codon, coding or genomic nucleotide alignment table for user-specified HLA alleles at user-specified positions.
</p>


<h3>Usage</h3>

<pre><code class="language-R">customAlign(alignType, alleles, positions)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>alignType</code></td>
<td>
<p>The type of alignment being searched. Allowed values are "prot", codon", "nuc" and "gen". Only one 'alignType' value is allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p>A vector of un-prefixed HLA allele names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>positions</code></td>
<td>
<p>Either a vector of variant positions, against which all loci will be aligned, or a list of vectors of nucleotide positions, exactly one vector for each allele, against which each corresponding allele will be aligned.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame of allele names and the corresponding nucleotide sequences for each desired nucleotide position. an error message is returned if input locus is not available in the ANHIG/IMGTHLA Github Repository.
</p>


<h3>Note</h3>

<p>This function requires that the HLAalignments object has been populated with alignments via the alignmentFull() function. C.f., customAlign("codon",c("DRB1<em>01:01","DQB1</em>02:01","DPB1<em>01:01"),c(1,2,3,7,8,9,13,14,15)) and customAlign("codon",c("DPB1</em>01:01:01:01","DQA1<em>01:01:01:01","DQB1</em>05:01:01:01"),list(19:35,1:4,6:9)).
</p>


</div>