<div class="container">

<table style="width: 100%;"><tr>
<td>validateMotif</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Determine if a Motif is Properly Formatted</h2>

<h3>Description</h3>

<p>Evaluates a motif to determine if the locus is valid and variants are valid.
</p>


<h3>Usage</h3>

<pre><code class="language-R">validateMotif(motif, alignType)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>motif</code></td>
<td>
<p>A character string identifying a sequence variant motif in the following format: Locus*#$~#$~#$, where ## identifies a variant position, and $ identifies the sequence variant. Both nucleotide and peptide motifs can be provided, and any number of variants can be specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alignType</code></td>
<td>
<p>A character string identifying the type of alignment being searched. Allowed values are "codon","gen", nuc" and "prot". Only one 'alignType' value is allowed.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>If the motif is valid, TRUE is returned. If the locus or body of the motif is invalid, FALSE is returned along with a brief message.
</p>


<h3>Examples</h3>

<pre><code class="language-R">validateMotif("A*-21M~2P","prot")
validateMotif("A*196G~301A~3046T","gen")

</code></pre>


</div>