<div class="container">

<table style="width: 100%;"><tr>
<td>multiLocusValidation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Apply validateLocus() to Multiple Loci</h2>

<h3>Description</h3>

<p>Applies validateLocus() to a vector of locus names, validates them against HLAgazeteer$gen, and returns a vector of validated locus names.
</p>


<h3>Usage</h3>

<pre><code class="language-R">multiLocusValidation(loci, source = "gDNA", verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>loci</code></td>
<td>
<p>A character vector of locus names found in the current HLAgazeteer.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>source</code></td>
<td>
<p>A character vector describing the type of alignment 'loci' should be validated against. Allowed options are 'AA' (protein), 'cDNA' (nucleotide) and 'gDNA' (genomic). Only a single value should be provided. The default value is 'gen'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>A logical value. If 'verbose' = TRUE, messages describing invalid 'loci' or 'source' values are generated. If 'verbose' = FALSE, no messages are generated.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A character vector of locus names that are present in HLAgazeteer$gen.
</p>


<h3>Note</h3>

<p>The results of this check should only be considered valid for the IPD-IMGT/HLA Database release version of the current HLAgazeteer.
</p>


<h3>Examples</h3>

<pre><code class="language-R">multiLocusValidation(loci = c("DRB1","DPB1","DQB8"))
multiLocusValidation(loci = c("A","B","C","D","Q"))

</code></pre>


</div>