<div class="container">

<table style="width: 100%;"><tr>
<td>HLAatlas</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Boundary Positions of Exons, Introns and UTRs in Amino Acid, cDNA and gDNA Alignments</h2>

<h3>Description</h3>

<p>A list object of sub-lists of R dataframes (atlases) for each locus with a protein (prot), cDNA (nuc), and gDNA (gen) alignment. Each atlas identifies the position of the exon (E), intron (I) or UTR (U) gene-feature boundaries in an alignment. Boundaries for non-standard hybrid (H), join (J), novel (N) and segment (S) features may be included in gene fragment and pseudogene atlases.
This object is built by the atlasFull() function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(HLAatlas)
</code></pre>


<h3>Format</h3>

<p>A list of 4 elements that include the gene features boundaries in each ANHIG/IMGTHLA sequence alignment.
</p>

<ul>
<li>
<p>(prot: peptide-alignment atlases)
</p>
</li>
<li>
<p>(nuc: cDNA-alignment atlases)
</p>
</li>
<li>
<p>(gen: gDNA-alignment atlases)
</p>
</li>
<li>
<p>(version: The IPD-IMGT/HLA Database release version under which these data were generated)
</p>
</li>
</ul>
<h3>Source</h3>

<p>https://github.com/ANHIG/IMGTHLA/tree/Latest/alignments
</p>


</div>