<div class="container">

<table style="width: 100%;"><tr>
<td>checkgDNAstart</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Identify gDNA Alignments in Which the First Feature Boundary is not Identified as Position +1.</h2>

<h3>Description</h3>

<p>Checks the position of the first feature boundary (the 5' UTR - Exon 1 boundary in expressed genes) in each gDNA alignment, and identifies those alignments in which that position is not 1.
</p>


<h3>Usage</h3>

<pre><code class="language-R">checkgDNAstart(verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>A logical indicating if loci with first feature boundary positions that are not 1 should be reported in the console (verbose = TRUE).</p>
</td>
</tr></table>
<h3>Details</h3>

<p>This function reviews the gDNA atlases in the HLAatlas object and returns a data frame of the first feature boundary position for each locus. For expressed genes and some pseudogenes, this is the position of the start of Exon 1.
</p>


<h3>Value</h3>

<p>A one-row data frame with one column for each locus with a gDNA alignment.
</p>


<h3>Note</h3>

<p>This function requires that the HLAalignments object has been populated with alignments via the alignmentFull() function.
</p>
<p>For internal HLAtools use.
</p>


</div>