<div class="container">

<table style="width: 100%;"><tr>
<td>Xoc.sevsample</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Likelihood-Ratio-Test Statistics: Several Sample Overdispersion Test Comparison</h2>

<h3>Description</h3>

<p>This routine provides the value of the likelihood-ratio-test statistic and the corresponding p-value to 
assess whether the overdispersion observed in multiple groups of microbiome samples are equal.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Xoc.sevsample(group.data, epsilon = 10^(-4))</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>group.data</code></td>
<td>
<p>A list where each element is a matrix of taxonomic counts(columns) for each sample(rows). (See Notes 1 and 2 in details)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>epsilon</code></td>
<td>
<p>Convergence tolerance. To terminate, the difference between two succeeding log-likelihoods must be smaller than epsilon. Default value is 10^(-4).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>To assess whether the over dispersion parameter vectors <code class="reqn">\theta_{\mathrm{m}}</code> observed in <code class="reqn">J</code> groups of microbiome samples are equal to each other, the following hypothesis 
<code class="reqn">\mathrm{H}_{\mathrm{o}}: \theta_{\mathrm{1}} = \cdots =\theta_{\mathrm{m}} =\cdots=\theta_{\mathrm{J}}=\theta_{\mathrm{o}}</code> versus <code class="reqn">\mathrm{H}_{\mathrm{a}}: \theta_{\mathrm{m}} \ne \theta_{\mathrm{o}}, m=1, \ldots, J</code> 
can be tested. In particular, the likelihood-ratio test statistic is used (Tvedebrink, 2010), which is given by,
</p>
<p style="text-align: center;"><code class="reqn">x_{\mathrm{oc}}=-2 \log\left\{\frac{L\left(\theta_{\mathrm{o}}; \mathbf{X}_{\mathrm{1}},\ldots, \mathbf{X}_{\mathrm{J}} \right)}{L\left(\theta_{\mathrm{1}},\ldots, \theta_{\mathrm{J}}; \mathbf{X}_{\mathrm{1}},\ldots, \mathbf{X}_{\mathrm{J}} \right)}\right\} .</code>
</p>
 
<p>The asymptotic null distribution of <code class="reqn">x_{\mathrm{oc}}</code> follows a Chi-square with degrees of freedom equal to (J-1) (Wilks, 1938).
</p>

<ol>
<li>
<p> Note 1: The matrices in <code>group.data</code> must contain the same taxa, in the same order.
</p>
</li>
<li>
<p> Note 2: Each taxa should be present in at least 1 sample, a column with all 0's may result in errors and/or invalid results.
</p>
</li>
</ol>
<h3>Value</h3>

<p>A list containing the Xoc statistics and p-value.</p>


<h3>References</h3>

<p>Tvedebrink, T. (2010). Overdispersion in allelic counts and theta-correction in forensic genetics. Theor Popul Biol 78, 200-210. <br>
Wilks, S. S. (1938). The Large-Sample Distribution of the Likelihood Ratio for Testing Composite Hypotheses. The Annals of Mathematical Statistics 9, 60-62.
</p>


<h3>Examples</h3>

<pre><code class="language-R">	data(saliva) 
	data(tonsils)
	
	### Combine the data sets into a single list
	group.data &lt;- list(saliva, tonsils)
	
	## Not run: 
		xoc &lt;- Xoc.sevsample(group.data)
		xoc
	
## End(Not run)
</code></pre>


</div>