<div class="container">

<table style="width: 100%;"><tr>
<td>HMP-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Hypothesis Testing and Power Calculations for Comparing Metagenomics Samples</h2>

<h3>Description</h3>

<p>This package provides tools for Generating data matrices following Multinomial and Dirichlet-Multinomial distributions, 
Computing the following test-statistics and their corresponding p-values, and Computing the power and size 
of the tests described above using Monte-Carlo simulations.
</p>


<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;">
		<code class="reqn">\textbf{Hypothesis Test}</code>	</td>
<td style="text-align: left;"> <code class="reqn">\textbf{Test Statistics Function}</code>	</td>
<td style="text-align: left;"> <code class="reqn">\textbf{Power Calculation Function}</code>	</td>
</tr>
<tr>
<td style="text-align: left;">
		2+ Sample Means w/ Reference Vector	</td>
<td style="text-align: left;"> <code>Xmc.sevsample</code>	</td>
<td style="text-align: left;"> <code>MC.Xmc.statistics</code> </td>
</tr>
<tr>
<td style="text-align: left;">
		1 Sample Mean w/ Reference Vector	</td>
<td style="text-align: left;"> <code>Xsc.onesample</code>	</td>
<td style="text-align: left;"> <code>MC.Xsc.statistics</code> </td>
</tr>
<tr>
<td style="text-align: left;">
		2+ Sample Means w/o Reference Vector	</td>
<td style="text-align: left;"> <code>Xmcupo.sevsample</code>	</td>
<td style="text-align: left;"> <code>MC.Xmcupo.statistics</code> </td>
</tr>
<tr>
<td style="text-align: left;">
		2+ Sample Overdispersions	</td>
<td style="text-align: left;"> <code>Xoc.sevsample</code>	</td>
<td style="text-align: left;"> <code>MC.Xoc.statistics</code> </td>
</tr>
<tr>
<td style="text-align: left;">
		2+ Sample DM-Distribution	</td>
<td style="text-align: left;"> <code>Xdc.sevsample</code>	</td>
<td style="text-align: left;"> <code>MC.Xdc.statistics</code>	</td>
</tr>
<tr>
<td style="text-align: left;">
		Multinomial vs DM	</td>
<td style="text-align: left;"> <code>C.alpha.multinomial</code>	</td>
<td style="text-align: left;"> <code>MC.ZT.statistics</code> </td>
</tr>
<tr>
<td style="text-align: left;">
	</td>
</tr>
</table>
<p>In addition to hypothesis testing and power calculations you can: <br></p>

<ol>
<li>
<p> Perform basic data management to exclude samples with fewer than pre-specified number of reads, 
collapse rare taxa and order the taxa by frequency. This is useful to exclude failed samples 
(i.e. samples with very few reads) - <code>Data.filter</code>
</p>
</li>
<li>
<p> Plot your data - <code>Barchart.data</code> 
</p>
</li>
<li>
<p> Generate random sample of Dirichlet Multinomial data with pre-specified parameters - <code>Dirichlet.multinomial</code>
</p>
</li>
</ol>
<p>Note: Thought the description of the functions refer its application to ranked abundance distributions (RAD) data, 
every function is also applicable to model species abundance data. See references for a discussion and application to both 
type of ecological data.
</p>


<h3>Author(s)</h3>

<p>Patricio S. La Rosa, Elena Deych, Berkley Shands, Sharina Carter, Dake Yang, William D. Shannon
</p>


<h3>References</h3>

<p>La Rosa PS, Brooks JP, Deych E, Boone EL, Edwards DJ, et al. (2012) Hypothesis Testing and Power Calculations for 
Taxonomic-Based Human Microbiome Data. PLoS ONE 7(12): e52078. doi:10.1371/journal.pone.0052078
</p>
<p>Yang D, Johnson J, Zhou X, Deych E, et al. (2019) Microbiome Recursive Partitioning. Currently Under Review.
</p>


</div>