<div class="container">

<table style="width: 100%;"><tr>
<td>HMPTrees-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Object Oriented Data Analysis of Taxonomic Trees</h2>

<h3>Description</h3>

<p>Object Oriented Data Analysis (OODA) methods to analyze Human Microbiome taxonomic trees directly. 
We provide tools to model, compare, and visualize populations of taxonomic trees.
</p>


<h3>Details</h3>

<p>HMP metagenomic sequences in a sample can be represented as a rooted taxonomic tree. Using supervised taxonomic methods a sequence is matched to a 
hierarchical taxa or taxonomy bins defined in a bacterial-taxonomy library such as, for example, the Ribosomal Database Project (RDP) (Cole, 2005). The 
supervised taxonomic analysis allows us to represent each sample (set of sequences) by a rooted taxonomic tree where the root corresponds to taxon at the 
Kingdom level, i.e., bacteria, and the leaves correspond to the taxa at the Genus level, and the width of the edges (paths) between taxonomic levels correspond 
to the 'abundances' of the descending taxon. <br></p>
<p>In particular, we combine RDP matches by adding RDP values of common taxon, which allows us to provide a measure 
of taxa abundance weighting on the confidence of each taxa assignment. The resulting taxonomic trees satisfy the following conditions: i) branches closer to 
the root have higher 'abundance' values than branches closer to leaves, and ii) the sum of the 'abundances' of all descending taxa under a common parent taxon 
cannot be larger than the 'abundance' of the corresponding parent taxon. <br></p>
<p>It is important to note that due to how the <code>ape</code> package works the following naming conventions apply to taxa names: 
</p>

<ol>
<li>
<p> Colons cannot be used in the taxa names at all.
</p>
</li>
<li>
<p> Each taxa name must be unique - you cannot have two seperate branches both have a child named 'unclassified' for example .
(We took the parent name and added a 'U' to the end to signify an unclassified in our data sets)
</p>
</li>
<li>
<p> There can only be one top level node. (Bacteria and Archaea cannot both exist unless there is an additional single level above them for example)
</p>
</li>
</ol>
<h3>Author(s)</h3>

<p>Patricio S. La Rosa, Elena Deych, Berkley Shands, William D. Shannon
</p>


<h3>References</h3>


<ol>
<li>
<p> Cole JR, Chai B, Farris RJ, Wang Q, Kulam SA, McGarrell DM, Garrity GM, Tiedje JM. The Ribosomal Database Project (RDP-II): 
sequences and tools for high-throughput rRNA analysis. Nucleic Acids Research 2005; 33: D294-D296.
</p>
</li>
<li>
<p> P. S. La Rosa, Berkley Shands, Elena Deych, Yanjiao Zhou, Erica Sodergren, George Weinstock, and William D. Shannon, 
"Object data analysis of taxonomic trees from human microbiome data,"PLoS ONE 7(11): e48996. doi:10.1371/journal.pone.0048996. Nov. 2012.
</p>
</li>
<li>
<p> Banks D, Constantine GM. Metric Models for Random Graphs. Journal of Classification 1998; 15: 199-223.
</p>
</li>
<li>
<p> Shannon WD, Banks D. Combining classification trees using MLE. Stat Med 1999; 18: 727-740.
</p>
</li>
</ol>
</div>