<div class="container">

<table style="width: 100%;"><tr>
<td>getGeneSeq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Extracting a gene sequence from NCBI database.
</h2>

<h3>Description</h3>

<p>This function retrieves a gene sequence from the NCBI database.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  getGeneSeq(chr, start, end, organism)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Chromosome number, numeric/string</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start</code></td>
<td>
<p>Start position, numeric</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>end</code></td>
<td>
<p>End position, numeric</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>organism</code></td>
<td>
<p>Name of the organism, string</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Extracting a gene sequence from NCBI database. For a list of available organism, visit
</p>
<p><a href="http://genome.ucsc.edu/cgi-bin/das/dsn">http://genome.ucsc.edu/cgi-bin/das/dsn</a>. All id="." field are available.
</p>


<h3>Value</h3>

<p>A string that contains the genomic sequence.
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Extracting for Sus Scrofa, build version 3:
getGeneSeq(1,2134,14532,"susScr3")
getGeneSeq(10,1233312,1233350,"hg38")

## End(Not run)
</code></pre>


</div>