<div class="container">

<table style="width: 100%;"><tr>
<td>autoTranslate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Attempt to automatically translate a gene list</h2>

<h3>Description</h3>

<p>Given a list of query gene list and a target gene list, the function
tries find the homology pairing that matches the query list to the target list. The query list
is a short list of genes while the target list is supposed to represent a large number of genes from the target
species. The default output will be the largest possible list. If <code>returnAllPossible = TRUE</code> then
all possible pairings with any matches are returned. It is possible to limit the
search by setting <code>possibleOrigins</code> and <code>possibleTargets</code>. Note that gene symbols of some species
are more similar to each other than others. Using this with small gene lists and without providing any
<code>possibleOrigins</code> or <code>possibleTargets</code> might return multiple hits, or if <code>returnAllPossible = TRUE</code>
a wrong match can be returned.
</p>


<h3>Usage</h3>

<pre><code class="language-R">autoTranslate(genes, targetGenes, possibleOrigins = NULL,
  possibleTargets = NULL, returnAllPossible = FALSE,
  db = homologene::homologeneData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>
<p>A list of genes to match the target. Symbols or NCBI ids</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>targetGenes</code></td>
<td>
<p>The target list. This list is supposed to represent a large number of genes
from the target species.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>possibleOrigins</code></td>
<td>
<p>Taxonomic identifiers of possible origin species</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>possibleTargets</code></td>
<td>
<p>Taxonomic identifiers of possible target species</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>returnAllPossible</code></td>
<td>
<p>if TRUE returns all possible pairings with non zero gene matches. If FALSE (default) returns the best match</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p>Homologene database to use.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame if <code>returnAllPossibe = FALSE</code> and a list of data frames if <code>TRUE</code>
</p>


</div>