<div class="container">

<table style="width: 100%;"><tr>
<td>CovMat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculation of covariance or correlation matrix</h2>

<h3>Description</h3>

<p>The theoretical covariance between pairs of markers is
calculated from either paternal haplotypes and maternal linkage
disequilibrium (LD) or vise versa. A genetic map is required. The
implementation relies on paternal half-sib families and biallelic markers
such as single nucleotide polymorphisms (SNP). If parental haplotypes are
incomplete (i.e., SNP alleles are missing), those parents will be discarded
at the corresponding pairs of SNPs. If maternal half-sib families are used,
the roles of sire/dam are swapped. Multiple families can be considered.
</p>


<h3>Usage</h3>

<pre><code class="language-R">CovMat(linkMat, haploMat, nfam, pos_chr, map_fun = "haldane", corr = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>linkMat</code></td>
<td>
<p>(p x p) matrix of maternal LD between pairs of p markers;
matrix is block diagonal in case of multiple chromosomes and must not
contain missing values; use zeros if LD is uncertain</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>haploMat</code></td>
<td>
<p>(2N x p) matrix of sires haplotypes for all chromosomes
(2 lines per sire); coding with 0's and 1's reflecting reference and
alternate alleles, respectively; missing values can be coded as NA or any
integer but not 0 and 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nfam</code></td>
<td>
<p>vector (LEN N) containing number of progeny per family or
scalar value in case of equal family size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos_chr</code></td>
<td>
<p>list (LEN number of chromosomes) of vectors (LEN number of
markers) of genetic positions in Morgan per chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map_fun</code></td>
<td>
<p>character string of mapping function used; so far "haldane"
(default) and "kosambi" are enabled</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>corr</code></td>
<td>
<p>logical; <code>FALSE</code> (default) if output is covariance matrix or
<code>TRUE</code> if output is correlation matrix</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>list (LEN 2) of matrix (DIM <code class="reqn">p1</code> x <code class="reqn">p1</code>) and vector
(LEN <code class="reqn">p1</code>) with <code class="reqn">p1 \le p</code>
</p>

<dl>
<dt><code>K</code></dt>
<dd>
<p>covariance matrix OR</p>
</dd>
<dt><code>R</code></dt>
<dd>
<p>correlation matrix</p>
</dd>
<dt><code>valid.snps</code></dt>
<dd>
<p>vector of SNP indices considered for covariance/
correlation matrix</p>
</dd>
</dl>
<h3>Note</h3>

<p>Family size is used for weighting covariance terms in case of multiple
half-sib families. It only matters if number of progeny differs.
</p>
<p>If maternal haplotypes (H.mothers) are used instead of maternal LD (matLD),
LD can be estimated from counting haplotype frequencies as:
</p>
<p><code>matLD &lt;- LDdam(inMat = H.mother, pos.chr)</code>
</p>
<p>If multiple chromosomes are considered, SNP positions are provided as, e.g.
</p>
<p><code>pos.chr &lt;- list(pos.snp.chr1, pos.snp.chr2, pos.snp.chr3)</code>
</p>


<h3>References</h3>

<p>Wittenburg, Bonk, Doschoris, Reyer (2020) Design of Experiments
for Fine-Mapping Quantitative Trait Loci in Livestock Populations. BMC
Genetics 21:66. doi: <a href="https://doi.org/10.1186/s12863-020-00871-1">10.1186/s12863-020-00871-1</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">  ### 1: INPUT DATA
  data(testdata)
  ### 2: COVARIANCE/CORRELATION MATRIX
  corrmat &lt;- CovMat(matLD, H.sire, 100, pos.chr, corr = TRUE)
  ### 3: TAGSNPS FROM CORRELATION MATRIX
  bin &lt;- tagSNP(corrmat$R)
  bin &lt;- tagSNP(corrmat$R, 0.5)
</code></pre>


</div>