<div class="container">

<table style="width: 100%;"><tr>
<td>LDdam</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculation of maternal LD matrix</h2>

<h3>Description</h3>

<p>Matrix containing linkage disequilibrium between marker pairs
on maternal gametes is set up by counting haplotypes frequencies.
</p>


<h3>Usage</h3>

<pre><code class="language-R">LDdam(inMat, pos_chr)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>inMat</code></td>
<td>
<p>[MATRIX] The maternal HAPLOTYPE matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos_chr</code></td>
<td>
<p>[LIST] The marker positions in Morgan on chromosomes.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function generates a block diagonal sparse matrix based on
Matrix::bdiag. Use as.matrix() to obtain a regular one.
</p>


<h3>Value</h3>


<dl>
<dt><code>Dd</code></dt>
<dd>
<p>(p x p) matrix of maternal LD</p>
</dd>
</dl>
<h3>Examples</h3>

<pre><code class="language-R"> ## haplotype matrix of n individuals at p SNPs
 p &lt;- 10; n &lt;- 4
 mat &lt;- matrix(ncol = p, nrow = 2 * n, sample(c(0, 1), size = 2 * n * p, replace = TRUE))
 LDdam(mat, list(1:p))
</code></pre>


</div>