<div class="container">

<table style="width: 100%;"><tr>
<td>tagSNP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>tagSNP</h2>

<h3>Description</h3>

<p>Grouping of markers depending on correlation structure
</p>


<h3>Usage</h3>

<pre><code class="language-R">tagSNP(mat, threshold = 0.8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mat</code></td>
<td>
<p>(p x p) correlation matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>lower value of correlation considered for grouping</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Grouping of markers is based on the  correlation matrix. Apart from
this, the strategy for grouping is similar to Carlson et al. (2004). A
representative marker is suggested for each group.
</p>


<h3>Value</h3>

<p>list (LEN number of groups) of lists (LEN 2); marker names correspond
to column names of mat
</p>

<dl>
<dt><code>snps</code></dt>
<dd>
<p>vector of marker IDs in group</p>
</dd>
<dt><code>tagsnp</code></dt>
<dd>
<p>representative marker suggested for this group</p>
</dd>
</dl>
<h3>References</h3>

<p>Carlson, Eberle, Rieder, Yi, Kruglyak &amp; Nickerson (2004)
Selecting a maximally informative set of single-nucleotide polymorphisms
for association analyses using linkage disequilibrium. American Journal of
Human Genetics 74:106-120.
</p>
<p>Wittenburg, Doschoris, Klosa (2021) Grouping of genomic markers in
populations with family structure BMC Bioinformatics 22:79.
doi: <a href="https://doi.org/10.1186/s12859-021-04010-0">10.1186/s12859-021-04010-0</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">  ### 1: INPUT DATA
  data(testdata)
  ### 2: COVARIANCE/CORRELATION MATRIX
  corrmat &lt;- CovMat(matLD, H.sire, 100, pos.chr, corr = TRUE)
  ### 3: TAGSNPS FROM CORRELATION MATRIX
  bin &lt;- tagSNP(corrmat$R)
  bin &lt;- tagSNP(corrmat$R, 0.5)
  as.numeric(unlist(rlist::list.select(bin, tagsnp)))
</code></pre>


</div>