<div class="container">

<table style="width: 100%;"><tr>
<td>readGenotype</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Read and Check the Genotype File</h2>

<h3>Description</h3>

<p>This function reads and checks genotype files.
</p>


<h3>Usage</h3>

<pre><code class="language-R">readGenotype(genotypePath, separatorGenotype = " ", check = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genotypePath</code></td>
<td>

<p><code>character</code> genotype path (animals (rows) and SNP (columns), SNP should be coded as 0, 1 and 2 for respectively AA, AB and BB.
Use 9 for missing data. please refer to vignette for more information)
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>separatorGenotype</code></td>
<td>

<p><code>character</code> separator character for genotype
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check</code></td>
<td>

<p><code>logical</code> check the genotype file for possible errors
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns the genotype matrix.
</p>


<h3>Note</h3>

<p>Please refer to vignette for more information.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># A comprehensive demo and example dataset is available from
# http://www-personal.une.edu.au/~cgondro2/hsphase.html
</code></pre>


</div>