<div class="container">

<table style="width: 100%;"><tr>
<td>ssp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Sire Imputation and Phasing
</h2>

<h3>Description</h3>

<p>Infer (impute) and phase sire's genotype based on the block structure matrix (recombination blocks) and homozygous sites of the half-sib genotype matrix.</p>


<h3>Usage</h3>

<pre><code class="language-R">ssp(blockMatrix, genotypeMatrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>blockMatrix</code></td>
<td>

<p><code>matrix</code> block structure (Output of <code>bmh</code>)
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotypeMatrix</code></td>
<td>

<p><code>matrix</code> half-sibs genotype
(each row includes the SNP of individuals, 0, 1 and 2 for respectively AA, AB and BB. Use 9 for missing data) </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a matrix (Imputed Sire) with two rows one for each haplotype of the sire (columns are SNP in the order of the genotype matrix). Alleles are coded as 0 (A) and 1 (B). Alleles that could not be imputed are coded as 9.
</p>


<h3>See Also</h3>

<p><code>phf</code>, <code>aio</code> and <code>imageplot</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">genotype &lt;- matrix(c(
0,2,1,1,1,
2,0,1,2,2,
2,2,1,0,2,
2,2,1,1,1,
0,0,2,1,0), ncol = 5, byrow = TRUE) # each row contains the SNP of individuals

(result &lt;- ssp(bmh(genotype), genotype)) 
</code></pre>


</div>