<div class="container">

<table style="width: 100%;"><tr>
<td>rpoh</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Reconstruct Pedigree Based on Matrix of Opposing Homozygotes</h2>

<h3>Description</h3>

<p>Reconstructs a half-sib pedigree based on a matrix of opposing homozygotes.</p>


<h3>Usage</h3>

<pre><code class="language-R">rpoh(genotypeMatrix, oh, forwardVectorSize = 30, excludeFP = TRUE, nsap = 3,
maxRec = 15, intercept = 26.3415, coefficient = 77.3171, snpnooh, method, maxsnpnooh)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genotypeMatrix</code></td>
<td>

<p><code>matrix</code> genotype of one chromosome (data should be numeric. Use 0, 1 and 2 for respectively AA, AB and BB. Use 9 for missing data)
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>oh</code></td>
<td>

<p><code>integer</code> Opposing homozygotes matrix (Output of <code>ohg</code>) </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>forwardVectorSize</code></td>
<td>

<p><code>integer</code> number of heterozygous sites used to validate recombination events or check for genotyping errors</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>excludeFP</code></td>
<td>

<p><code>logical</code> excludes SNPs that may cause heterozygous sites in the sire due to genotyping errors or map errors</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsap</code></td>
<td>

<p><code>integer</code> number of SNP per block to validate recombinations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxRec</code></td>
<td>

<p><code>integer</code> maximum number of expected recombinations per individual</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>intercept</code></td>
<td>

<p><code>integer</code> intercept of fitted model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coefficient</code></td>
<td>

<p><code>integer</code> coefficient of fitted model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpnooh</code></td>
<td>

<p><code>integer</code> number of SNPs used to create <em>oh</em> matrix (this number must be divided by 1000)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>

<p><code>character</code> pedigree reconstruction method</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxsnpnooh</code></td>
<td>

<p><code>numeric</code> the maximum number of allowing opposing homozygote in a half-sib family
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Four methods <em>simple</em>, <em>recombinations</em>, <em>calus</em> and <em>manual</em> can be
utilized to reconstruct the pedigree. 
<br><br>
The following examples show the arguments require for each method.
</p>
<p>pedigree1 &lt;- rpoh(oh = oh, snpnooh = 732, method = "simple") <br>
pedigree2 &lt;- rpoh(genotypeMatrix = genotypeChr1, oh = ohg(genotype), maxRec = 10 , method = "recombinations")  <br>
pedigree3 &lt;- rpoh(genotypeMatrix = genotype, oh = oh, method = "calus") <br>
pedigree4 &lt;- rpoh(oh = oh, maxsnpnooh = 31662, method = "manual")
<br></p>


<h3>Value</h3>

<p>Returns a data frame with two columns, the first column is animals' ID and the second column is sire identifiers (randomly generated).
</p>


<h3>Note</h3>

<p>Method can be <em>recombinations</em>, <em>simple</em>, <em>calus</em> or <em>manual</em>. Please refer to vignette for more information.<br></p>
<p>The sire genotype should be removed before using this function utilizing  <code>pogc</code> function.
</p>


<h3>See Also</h3>

<p><code>bmh</code> and <code>recombinations</code> 
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Please run demo(hsphase)
</code></pre>


</div>