<div class="container">

<table style="width: 100%;"><tr>
<td>impute</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Impute of Low Density SNP Marker to High Density (Paternal Strand)</h2>

<h3>Description</h3>

<p>Impute the paternal strand from low density to high density utilising high density sire haplotype.</p>


<h3>Usage</h3>

<pre><code class="language-R">impute(halfsib_genotype_ld, sire_hd, bmh_forwardVectorSize = 30, 
bmh_excludeFP = TRUE, bmh_nsap = 3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>halfsib_genotype_ld</code></td>
<td>
<p><code>matrix</code>
half-sib genotypes with low density marker (one half-sib per row, with SNP ordered by mapping position in the columns.
Data should be numeric. Use 0, 1 and 2 respectively for AA, AB and BB. Use 9 for missing data)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sire_hd</code></td>
<td>
<p><code>matrix</code>
haplotype of sire (this parameter can be sequence data or any phased sire - the matrix should have rownames which are the sample
IDs and colnames which are the SNP names)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bmh_forwardVectorSize</code></td>
<td>

<p><code>integer</code> number of heterozygous sites used to validate recombination events or check for genotyping errors</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bmh_excludeFP</code></td>
<td>

<p><code>logical</code> exclude SNPs that may cause heterozygous sites in the sire due to genotyping errors or map errors</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bmh_nsap</code></td>
<td>

<p><code>integer</code> number of SNPs per block</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Return an imputed half-sib matrix. 
</p>


<h3>See Also</h3>

<p><code>bmh</code>,  <code>ssp</code> and  <code>phf</code>
</p>


</div>