<div class="container">

<table style="width: 100%;"><tr>
<td>makehaplist</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Make list of sire haplotypes</h2>

<h3>Description</h3>

<p>List of sire haplotypes is set up in the format required for
hsrecombi. Haplotypes (obtained by external software) are provided.
</p>


<h3>Usage</h3>

<pre><code class="language-R">makehaplist(daughterSire, hapSire, nmin = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>daughterSire</code></td>
<td>
<p>vector (LEN n) of sire ID for each progeny</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hapSire</code></td>
<td>
<p>matrix (DIM 2N x p + 1) of sire haplotype at p SNPs; 2 lines
per sire, 1. columns contains sire ID</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nmin</code></td>
<td>
<p>scalar, minimum number of progeny required, default 1</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>list (LEN 2) of lists. For each sire:
</p>

<dl>
<dt><code>famID</code></dt>
<dd>
<p>list (LEN N) of vectors (LEN n.progeny) of progeny
indices relating to lines in genotype matrix</p>
</dd>
<dt><code>sireHap</code></dt>
<dd>
<p>list (LEN N) of matrices (DIM 2 x p) of sire
haplotypes (0, 1) on investigated chromosome</p>
</dd>
</dl>
<h3>Examples</h3>

<pre><code class="language-R">  data(targetregion)
  hap &lt;- makehaplist(daughterSire, hapSire)
</code></pre>


</div>