<div class="container">

<table style="width: 100%;"><tr>
<td>editraw</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Editing results of hsrecombi</h2>

<h3>Description</h3>

<p>Process raw results from <code>hsrecombi</code>, decide which out of
two sets of estimates is more likely and prepare list of final results
</p>


<h3>Usage</h3>

<pre><code class="language-R">editraw(Roh, map1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Roh</code></td>
<td>
<p>list of raw results from <code>hsrecombi</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map1</code></td>
<td>
<p>data.frame containing information on physical map, at least:
</p>

<dl>
<dt><code>SNP</code></dt>
<dd>
<p>SNP ID</p>
</dd>
<dt><code>locus_Mb</code></dt>
<dd>
<p>physical position in Mbp of SNP on chromosomes</p>
</dd>
<dt><code>Chr</code></dt>
<dd>
<p>chromosome of SNP</p>
</dd>
</dl>
</td>
</tr>
</table>
<h3>Value</h3>

<p>final table of results
</p>

<dl>
<dt><code>SNP1</code></dt>
<dd>
<p>index 1. SNP</p>
</dd>
<dt><code>SNP2</code></dt>
<dd>
<p>index 2. SNP</p>
</dd>
<dt><code>D</code></dt>
<dd>
<p>maternal LD</p>
</dd>
<dt><code>fAA</code></dt>
<dd>
<p>frequency of maternal haplotype 1-1</p>
</dd>
<dt><code>fAB</code></dt>
<dd>
<p>frequency of maternal haplotype 1-0</p>
</dd>
<dt><code>fBA</code></dt>
<dd>
<p>frequency of maternal haplotype 0-1</p>
</dd>
<dt><code>fBB</code></dt>
<dd>
<p>frequency of maternal haplotype 0-0</p>
</dd>
<dt><code>p1</code></dt>
<dd>
<p>Maternal allele frequency (allele 1) at <code>SNP1</code></p>
</dd>
<dt><code>p2</code></dt>
<dd>
<p>Maternal allele frequency (allele 1) at <code>SNP2</code></p>
</dd>
<dt><code>nfam1</code></dt>
<dd>
<p>size of genomic family 1</p>
</dd>
<dt><code>nfam2</code></dt>
<dd>
<p>size of genomic family 2</p>
</dd>
<dt><code>error</code></dt>
<dd>
<p>0 if computations were without error; 1 if EM algorithm
did not converge</p>
</dd>
<dt><code>iteration</code></dt>
<dd>
<p>number of EM iterations</p>
</dd>
<dt><code>theta</code></dt>
<dd>
<p>paternal recombination rate</p>
</dd>
<dt><code>r2</code></dt>
<dd>
<p><code class="reqn">r^2</code> of maternal LD</p>
</dd>
<dt><code>logL</code></dt>
<dd>
<p>value of log likelihood function</p>
</dd>
<dt><code>unimodal</code></dt>
<dd>
<p>1 if likelihood is unimodal; 0 if likelihood is
bimodal</p>
</dd>
<dt><code>critical</code></dt>
<dd>
<p>0 if parameter estimates were unique; 1 if parameter
estimates were obtained via decision process</p>
</dd>
<dt><code>locus_Mb</code></dt>
<dd>
<p>physical distance between SNPs in Mbp</p>
</dd>
</dl>
<h3>Examples</h3>

<pre><code class="language-R">  ### test data
  data(targetregion)
  ### make list for paternal half-sib families
  hap &lt;- makehaplist(daughterSire, hapSire)
  ### parameter estimates on a chromosome
  res &lt;- hsrecombi(hap, genotype.chr)
  ### post-processing to achieve final and valid set of estimates
  final &lt;- editraw(res, map.chr)
</code></pre>


</div>