<div class="container">

<table style="width: 100%;"><tr>
<td>hsrecombi</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimation of recombination rate and maternal LD</h2>

<h3>Description</h3>

<p>Wrapper function for estimating recombination rate and maternal
linkage disequilibrium between intra-chromosomal SNP pairs by calling EM
algorithm
</p>


<h3>Usage</h3>

<pre><code class="language-R">hsrecombi(hap, genotype.chr, exclude = NULL, only.adj = FALSE, prec = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>hap</code></td>
<td>
<p>list (LEN 2) of lists
</p>

<dl>
<dt>famID</dt>
<dd>
<p>list (LEN number of sires) of vectors (LEN n.progeny) of
progeny indices relating to lines in genotype matrix</p>
</dd>
<dt>sireHap</dt>
<dd>
<p>list (LEN number of sires) of matrices (DIM 2 x p)
of sire haplotypes (0, 1) on investigated chromosome</p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotype.chr</code></td>
<td>
<p>matrix (DIM n x p) of all progeny genotypes (0, 1, 2) on
a chromosome with p SNPs; 9 indicates missing genotype</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exclude</code></td>
<td>
<p>vector (LEN &lt; p) of SNP IDs (for filtering column names of
<code>genotype.chr)</code> to be excluded from analysis (default NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>only.adj</code></td>
<td>
<p>logical; if <code>TRUE</code>, recombination rate is calculated
only between neighbouring markers</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prec</code></td>
<td>
<p>scalar; precision of estimation</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Paternal recombination rate and maternal linkage disequilibrium (LD)
are estimated for pairs of biallelic markers (such as single nucleotide
polymorphisms; SNPs) from progeny genotypes and sire haplotypes. At least
one sire has to be double heterozygous at the investigated pairs of SNPs.
All progeny are merged in two genomic families: (1) coupling phase family
if sires are double heterozygous 0-0/1-1 and (2) repulsion phase family if
sires are double heterozygous 0-1/1-0.
So far it is recommended processing the chromosomes separately.
If maternal half-sib families are used, the roles of sire/dam are swapped.
Multiple families can be considered.
</p>


<h3>Value</h3>

<p>list (LEN p - 1) of data.frames; for each SNP, parameters are
estimated with all following SNPs; two solutions (prefix sln1 and sln2) are
obtained for two runs of the EM algorithm
</p>

<dl>
<dt><code>SNP1</code></dt>
<dd>
<p>ID of 1. SNP</p>
</dd>
<dt><code>SNP2</code></dt>
<dd>
<p>ID of 2. SNP</p>
</dd>
<dt><code>D</code></dt>
<dd>
<p>maternal LD</p>
</dd>
<dt><code>fAA</code></dt>
<dd>
<p>frequency of maternal haplotype 1-1</p>
</dd>
<dt><code>fAB</code></dt>
<dd>
<p>frequency of maternal haplotype 1-0</p>
</dd>
<dt><code>fBA</code></dt>
<dd>
<p>frequency of maternal haplotype 0-1</p>
</dd>
<dt><code>fBB</code></dt>
<dd>
<p>frequency of maternal haplotype 0-0</p>
</dd>
<dt><code>p1</code></dt>
<dd>
<p>Maternal allele frequency (allele 1) at <code>SNP1</code></p>
</dd>
<dt><code>p2</code></dt>
<dd>
<p>Maternal allele frequency (allele 1) at <code>SNP2</code></p>
</dd>
<dt><code>nfam1</code></dt>
<dd>
<p>size of genomic family 1</p>
</dd>
<dt><code>nfam2</code></dt>
<dd>
<p>size of genomic family 2</p>
</dd>
<dt><code>error</code></dt>
<dd>
<p>0 if computations were without error; 1 if EM algorithm
did not converge</p>
</dd>
<dt><code>iteration</code></dt>
<dd>
<p>number of EM iterations</p>
</dd>
<dt><code>theta</code></dt>
<dd>
<p>paternal recombination rate</p>
</dd>
<dt><code>r2</code></dt>
<dd>
<p><code class="reqn">r^2</code> of maternal LD</p>
</dd>
<dt><code>logL</code></dt>
<dd>
<p>value of log likelihood function</p>
</dd>
<dt><code>unimodal</code></dt>
<dd>
<p>1 if likelihood is unimodal; 0 if likelihood is
bimodal</p>
</dd>
<dt><code>critical</code></dt>
<dd>
<p>0 if parameter estimates are unique; 1 if parameter
estimates at both solutions are valid, then decision process follows in
post-processing function "editraw"</p>
</dd>
</dl>
<p>Afterwards, solutions are compared and processed with function
<code>editraw</code>, yielding the final estimates for each valid pair of SNPs.
</p>


<h3>References</h3>

<p>Hampel, A., Teuscher, F., Gomez-Raya, L., Doschoris, M. &amp; Wittenburg, D.
(2018) Estimation of recombination rate and maternal linkage
disequilibrium in half-sibs. Frontiers in Genetics 9:186.
doi: <a href="https://doi.org/10.3389/fgene.2018.00186">10.3389/fgene.2018.00186</a>
</p>
<p>Gomez-Raya, L. (2012) Maximum likelihood estimation of linkage
disequilibrium in half-sib families. Genetics 191:195-213.
</p>


<h3>Examples</h3>

<pre><code class="language-R">  ### test data
  data(targetregion)
  ### make list for paternal half-sib families
  hap &lt;- makehaplist(daughterSire, hapSire)
  ### parameter estimates on a chromosome
  res &lt;- hsrecombi(hap, genotype.chr)
  ### post-processing to achieve final and valid set of estimates
  final &lt;- editraw(res, map.chr)
</code></pre>


</div>