<div class="container">

<table style="width: 100%;"><tr>
<td>geneticPosition</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimation of genetic position</h2>

<h3>Description</h3>

<p>Estimation of genetic positions (in centi Morgan)
</p>


<h3>Usage</h3>

<pre><code class="language-R">geneticPosition(final, map1, exclude = NULL, threshold = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>final</code></td>
<td>
<p>table of results produced by <code>editraw</code> with pairwise
estimates of recombination rate between p SNPs within chromosome; minimum
required data frame with columns <code>SNP1</code>, <code>SNP2</code> and <code>theta</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map1</code></td>
<td>
<p>data.frame containing information on physical map, at least:
</p>

<dl>
<dt><code>SNP</code></dt>
<dd>
<p>SNP ID</p>
</dd>
<dt><code>locus_Mb</code></dt>
<dd>
<p>physical position in Mbp of SNP on chromosomes</p>
</dd>
<dt><code>Chr</code></dt>
<dd>
<p>chromosome of SNP</p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exclude</code></td>
<td>
<p>optional vector (LEN &lt; p) of SNP IDs to be excluded (e.g.,
candidates of misplaced SNPs; default NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>optional value; recombination rates &lt;= threshold are
considered for smoothing approach assuming theta ~ Morgan (default 0.05)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Smoothing of recombination rates (theta) &lt;= 0.05 via quadratic
optimization provides an approximation of genetic distances (in Morgan)
between SNPs. The cumulative sum * 100 yields the genetic positions in cM.
</p>
<p>The minimization problem <code>(theta - D d)^2</code> is solved s.t. d &gt; 0 where
d is the vector of genetic distances between adjacent markers but theta is
not restricted to adjacent markers. The incidence matrix D contains 1's for
those intervals contributing to the total distance relevant for each theta.
</p>
<p>Estimates of theta = 1e-6 are neglected as these values coincide with start
values and indicate that (because of a very flat likelihood surface) no
meaningful estimate of recombination rate has been obtained.
</p>


<h3>Value</h3>

<p>list (LEN 2)
</p>

<dl>
<dt>gen.cM</dt>
<dd>
<p>vector (LEN p) of genetic positions of SNPs (in cM)</p>
</dd>
<dt>gen.Mb</dt>
<dd>
<p>vector (LEN p) of physical positions of SNPs (in Mbp)</p>
</dd>
</dl>
<h3>References</h3>

<p>Qanbari, S. &amp; Wittenburg, D. (2020) Male recombination map of the autosomal
genome in German Holstein. Genetics Selection Evolution 52:73.
doi: <a href="https://doi.org/10.1186/s12711-020-00593-z">10.1186/s12711-020-00593-z</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">  ### test data
  data(targetregion)
  ### make list for paternal half-sib families
  hap &lt;- makehaplist(daughterSire, hapSire)
  ### parameter estimates on a chromosome
  res &lt;- hsrecombi(hap, genotype.chr)
  ### post-processing to achieve final and valid set of estimates
  final &lt;- editraw(res, map.chr)
  ### approximation of genetic positions
  pos &lt;- geneticPosition(final, map.chr)
</code></pre>


</div>