<div class="container">

<table style="width: 100%;"><tr>
<td>htlr_fit</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Fit a HTLR Model (Internal API)</h2>

<h3>Description</h3>

<p>This function trains linear logistic regression models with HMC in restricted Gibbs sampling.
It also makes predictions for test cases if <code>X_ts</code> are provided.
</p>


<h3>Usage</h3>

<pre><code class="language-R">htlr_fit(
  X_tr,
  y_tr,
  fsel = 1:ncol(X_tr),
  stdzx = TRUE,
  ptype = c("t", "ghs", "neg"),
  sigmab0 = 2000,
  alpha = 1,
  s = -10,
  eta = 0,
  iters_h = 1000,
  iters_rmc = 1000,
  thin = 1,
  leap_L = 50,
  leap_L_h = 5,
  leap_step = 0.3,
  hmc_sgmcut = 0.05,
  initial_state = "lasso",
  keep.warmup.hist = FALSE,
  silence = TRUE,
  rep.legacy = TRUE,
  alpha.rda = 0.2,
  lasso.lambda = seq(0.05, 0.01, by = -0.01),
  X_ts = NULL,
  predburn = NULL,
  predthin = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X_tr</code></td>
<td>
<p>Input matrix, of dimension nobs by nvars; each row is an observation vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y_tr</code></td>
<td>
<p>Vector of response variables. Must be coded as non-negative integers, 
e.g., 1,2,...,C for C classes, label 0 is also allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fsel</code></td>
<td>
<p>Subsets of features selected before fitting, such as by univariate screening.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stdzx</code></td>
<td>
<p>Logical; if <code>TRUE</code>, the original feature values are standardized to have <code>mean = 0</code> 
and <code>sd = 1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ptype</code></td>
<td>
<p>The prior to be applied to the model. Either "t" (student-t, default), 
"ghs" (horseshoe), or "neg" (normal-exponential-gamma).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigmab0</code></td>
<td>
<p>The <code>sd</code> of the normal prior for the intercept.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>The degree freedom of t/ghs/neg prior for coefficients.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>s</code></td>
<td>
<p>The log scale of priors (logw) for coefficients.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eta</code></td>
<td>
<p>The <code>sd</code> of the normal prior for logw. When it is set to 0, logw is fixed. 
Otherwise, logw is assigned with a normal prior and it will be updated during sampling.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iters_h</code></td>
<td>
<p>A positive integer specifying the number of warmup (aka burnin).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iters_rmc</code></td>
<td>
<p>A positive integer specifying the number of iterations after warmup.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thin</code></td>
<td>
<p>A positive integer specifying the period for saving samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>leap_L</code></td>
<td>
<p>The length of leapfrog trajectory in sampling phase.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>leap_L_h</code></td>
<td>
<p>The length of leapfrog trajectory in burnin phase.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>leap_step</code></td>
<td>
<p>The stepsize adjustment multiplied to the second-order partial derivatives of log posterior.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hmc_sgmcut</code></td>
<td>
<p>The coefficients smaller than this criteria will be fixed in 
each HMC updating step.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>initial_state</code></td>
<td>
<p>The initial state of Markov Chain; can be a previously 
fitted <code>fithtlr</code> object, or a user supplied initial state vector, or 
a character string matches the following:  
</p>

<ul>
<li>
<p> "lasso" - (Default) Use Lasso initial state with <code>lambda</code> chosen by 
cross-validation. Users may specify their own candidate <code>lambda</code> values via 
optional argument <code>lasso.lambda</code>. Further customized Lasso initial 
states can be generated by <code>lasso_deltas</code>.    
</p>
</li>
<li>
<p> "bcbcsfrda" - Use initial state generated by package <code>BCBCSF</code> 
(Bias-corrected Bayesian classification). Further customized BCBCSF initial 
states can be generated by <code>bcbcsf_deltas</code>. WARNING: This type of 
initial states can be used for continuous features such as gene expression profiles, 
but it should not be used for categorical features such as SNP profiles.
</p>
</li>
<li>
<p> "random" - Use random initial values sampled from N(0, 1).     
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep.warmup.hist</code></td>
<td>
<p>Warmup iterations are not recorded by default, set <code>TRUE</code> to enable it.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>silence</code></td>
<td>
<p>Setting it to <code>FALSE</code> for tracking MCMC sampling iterations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rep.legacy</code></td>
<td>
<p>Logical; if <code>TRUE</code>, the output produced in <code>HTLR</code> versions up to 
legacy-3.1-1 is reproduced. The speed would be typically slower than non-legacy mode on
multi-core machine.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha.rda</code></td>
<td>
<p>A user supplied alpha value for <code>bcbcsf_deltas</code> when
setting up BCBCSF initial state. Default: 0.2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lasso.lambda</code></td>
<td>
<p>- A user supplied lambda sequence for <code>lasso_deltas</code> when 
setting up Lasso initial state. Default: {.01, .02, ..., .05}. Will be ignored if 
<code>rep.legacy</code> is set to <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X_ts</code></td>
<td>
<p>Test data which predictions are to be made.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>predburn, predthin</code></td>
<td>
<p>For prediction base on <code>X_ts</code> (when supplied), <code>predburn</code> of 
Markov chain (super)iterations will be discarded, and only every <code>predthin</code> are used for inference.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of fitting results. If <code>X_ts</code> is not provided, the list is an object 
with S3 class <code>htlr.fit</code>.
</p>


<h3>References</h3>

<p>Longhai Li and Weixin Yao (2018). Fully Bayesian Logistic Regression 
with Hyper-Lasso Priors for High-dimensional Feature Selection.
<em>Journal of Statistical Computation and Simulation</em> 2018, 88:14, 2827-2851.
</p>


</div>