<div class="container">

<table style="width: 100%;"><tr>
<td>do_cv_direct</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Direct HTRX: k-fold cross-validation on short haplotypes</h2>

<h3>Description</h3>

<p>Direct k-fold cross-validation used to compute the out-of-sample variance explained by selected features from HTRX.
It can be applied to select haplotypes based on HTR, or select single nucleotide polymorphisms (SNPs).
</p>


<h3>Usage</h3>

<pre><code class="language-R">do_cv_direct(
  data_nosnp,
  featuredata,
  featurecap = dim(featuredata)[2],
  usebinary = 1,
  method = "simple",
  criteria = "BIC",
  gain = TRUE,
  runparallel = FALSE,
  mc.cores = 6,
  fold = 10,
  kfoldseed = 123,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data_nosnp</code></td>
<td>
<p>a data frame with outcome (the outcome must be the first column),
fixed covariates (for example, sex, age and the first 18 PCs) if there are,
and without SNPs or haplotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>featuredata</code></td>
<td>
<p>a data frame of the feature data, e.g. haplotype data created by HTRX or SNPs.
These features exclude all the data in <code>data_nosnp</code>, and will be selected using 2-step cross-validation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>featurecap</code></td>
<td>
<p>a positive integer which manually sets the maximum number of independent features.
By default, <code>featurecap=40</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>usebinary</code></td>
<td>
<p>a non-negative number representing different models.
Use linear model if <code>usebinary=0</code>,
use logistic regression model via <code>fastglm</code> if <code>usebinary=1</code> (by default),
and use logistic regression model via <code>glm</code> if <code>usebinary&gt;1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>the method used for data splitting, either <code>"simple"</code> (default) or <code>"stratified"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>criteria</code></td>
<td>
<p>the criteria for model selection, either <code>"BIC"</code> (default), <code>"AIC"</code> or <code>"lasso"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gain</code></td>
<td>
<p>logical. If <code>gain=TRUE</code> (default), report the variance explained in addition to fixed covariates;
otherwise, report the total variance explained by all the variables.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>runparallel</code></td>
<td>
<p>logical. Use parallel programming based on <code>mclapply</code> function from R package <code>"parallel"</code> or not.
Note that for Windows users, <code>mclapply</code> doesn't work, so please set <code>runparallel=FALSE</code> (default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mc.cores</code></td>
<td>
<p>an integer giving the number of cores used for parallel programming.
By default, <code>mc.cores=6</code>.
This only works when <code>runparallel=TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fold</code></td>
<td>
<p>a positive integer specifying how many folds
the data should be split into for cross-validation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kfoldseed</code></td>
<td>
<p>a positive integer specifying the seed used to
split data for k-fold cross validation. By default, <code>kfoldseed=123</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical. If <code>verbose=TRUE</code>, print out the inference steps. By default, <code>verbose=FALSE</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Function <code>do_cv_direct</code> directly performs k-fold cross-validation: features are
selected from the training set using a specified <code>criteria</code>,
and the out-of-sample variance explained by the selected features are computed on the test set.
This function runs faster than <code>do_cv</code> with large <code>sim_times</code>, but may lose
some accuracy, and it doesn't return a fixed set of features.
</p>


<h3>Value</h3>

<p><code>do_cv_direct</code> returns a list of the out-of-sample variance explained in each of the test set,
and the features selected in each of the k training sets.
</p>


<h3>References</h3>

<p>Yang Y, Lawson DJ. HTRX: an R package for learning non-contiguous haplotypes associated with a phenotype. Bioinformatics Advances 3.1 (2023): vbad038.
</p>
<p>Barrie, W., Yang, Y., Irving-Pease, E.K. et al. Elevated genetic risk for multiple sclerosis emerged in steppe pastoralist populations. Nature 625, 321â€“328 (2024).
</p>
<p>Eforn, B. "Bootstrap methods: another look at the jackknife." The Annals of Statistics 7 (1979): 1-26.
</p>
<p>Schwarz, Gideon. "Estimating the dimension of a model." The annals of statistics (1978): 461-464.
</p>
<p>McFadden, Daniel. "Conditional logit analysis of qualitative choice behavior." (1973).
</p>
<p>Akaike, Hirotugu. "A new look at the statistical model identification." IEEE transactions on automatic control 19.6 (1974): 716-723.
</p>
<p>Tibshirani, Robert. "Regression shrinkage and selection via the lasso." Journal of the Royal Statistical Society: Series B (Methodological) 58.1 (1996): 267-288.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## use dataset "example_hap1", "example_hap2" and "example_data_nosnp"
## "example_hap1" and "example_hap2" are
## both genomes of 8 SNPs for 5,000 individuals (diploid data)
## "example_data_nosnp" is an example dataset
## which contains the outcome (binary), sex, age and 18 PCs

## visualise the covariates data
## we will use only the first two covariates: sex and age in the example
head(HTRX::example_data_nosnp)

## visualise the genotype data for the first genome
head(HTRX::example_hap1)

## we perform HTRX on the first 4 SNPs
## we first generate all the haplotype data, as defined by HTRX
HTRX_matrix=make_htrx(HTRX::example_hap1[,1:4],
                      HTRX::example_hap2[,1:4])

## If the data is haploid, please set
## HTRX_matrix=make_htrx(HTRX::example_hap1[,1:4],
##                       HTRX::example_hap1[,1:4])

## next compute the maximum number of independent features
featurecap=htrx_max(nsnp=4,cap=10)
## then perform HTRX using direct cross-validation
## If we want to compute the total variance explained
## we can set gain=FALSE in the above example

htrx_results &lt;- do_cv_direct(HTRX::example_data_nosnp[,1:3],
                             HTRX_matrix,featurecap=featurecap,
                             usebinary=1,method="stratified",
                             criteria="lasso",gain=TRUE,
                             runparallel=FALSE,verbose=TRUE)

</code></pre>


</div>