<div class="container">

<table style="width: 100%;"><tr>
<td>htrx_max</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Maximum independent features for HTRX</h2>

<h3>Description</h3>

<p>The maximum number of independent features in principle from
haplotypes (i.e. interactions between SNPs) generated by Haplotype Trend Regression with eXtra flexibility (HTRX).
</p>


<h3>Usage</h3>

<pre><code class="language-R">htrx_max(nsnp, n_haps = NULL, cap = 40, max_int = NULL, htr = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nsnp</code></td>
<td>
<p>a positive integer giving the number of single nucleotide polymorphisms (SNPs) included in the haplotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_haps</code></td>
<td>
<p>a positive integer giving the number of haplotypes,
which is also the number of columns of the HTRX or HTR matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cap</code></td>
<td>
<p>a positive integer which manually sets the maximum number of independent features.
By default, <code>cap=40</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_int</code></td>
<td>
<p>a positive integer which specifies the maximum number of SNPs that can interact.
If no value is given (by default), interactions between all the SNPs will be considered.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>htr</code></td>
<td>
<p>logical. If <code>htr=TRUE</code>, the functions returns the maximum number of independent
features for HTR. By default, <code>htr=FALSE</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The maximum number of independent features in principle is 2<sup>nsnp</sup>-1
for haplotypes containing interactions between all different numbers of SNPs.
However, if <code>max_int &lt; nsnp</code>, i.e. only the interactions between at most <code>max_int</code> SNPs are investigated,
there will be fewer maximum number of independent features.
You can also manually set the upper limit of independent features (by setting <code>cap</code>) that can be included in the final HTRX or HTR model.
</p>


<h3>Value</h3>

<p><code>htrx_max</code> returns a positive integer giving the maximum
number of independent features to be included in the analysis.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## the maximum number of independent haplotypes consisted of 4 SNPs from HTRX
htrx_max(nsnp=4,n_haps=(3^4-1))
</code></pre>


</div>