<div class="container">

<table style="width: 100%;"><tr>
<td>htrx_nfeatures</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Total number of features for HTRX</h2>

<h3>Description</h3>

<p>The total number of features in principle from haplotypes (i.e. interactions between SNPs)
generated by Haplotype Trend Regression with eXtra flexibility (HTRX) .
</p>


<h3>Usage</h3>

<pre><code class="language-R">htrx_nfeatures(nsnp, max_int = NULL, htr = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nsnp</code></td>
<td>
<p>a positive integer giving the number of single nucleotide polymorphisms (SNPs) included in the haplotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_int</code></td>
<td>
<p>a positive integer which specifies the maximum number of SNPs that can interact.
If no value is given (by default), interactions between all the SNPs will be considered.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>htr</code></td>
<td>
<p>logical. If <code>htr=TRUE</code>, the function returns the total number of
features for HTR. By default, <code>htr=FALSE</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The total number of features in principle is 3<sup>nsnp</sup>-1
for haplotypes containing interactions between all different numbers of SNPs.
However, if <code>max_int &lt; nsnp</code>, i.e. only the interactions between at most <code>max_int</code> SNPs are investigated,
there will be fewer total number of features.
</p>


<h3>Value</h3>

<p><code>htrx_nfeatures</code> returns a positive integer giving the total number
of features that each analysis includes.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## the total number of haplotypes consisted of 6 SNPs
## for at most 3-SNP interactions
htrx_nfeatures(nsnp=6,max_int=3)
</code></pre>


</div>