<div class="container">

<table style="width: 100%;"><tr>
<td>HTRX-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>HTRX: Haplotype Trend Regression with eXtra flexibility</h2>

<h3>Description</h3>

<p>This is the software for "HTRX - Haplotype Trend Regression with eXtra flexibility (HTRX)"
based on the papar Genetic risk for Multiple Sclerosis originated in Pastoralist Steppe populations, Barrie W, Yang Y, Attfield K E, et al (2022).
</p>
<p>HTRX searches for haplotype patterns that include single nucleotide polymorphisms (SNPs) and non-contiguous haplotypes.
</p>
<p>HTRX is a template gives a value for each SNP taking values of ‘0’ or ‘1’, reflecting
whether the reference allele of each SNP is present or absent, or an ‘X’ meaning either value is allowed.
</p>
<p>We used a two-step procedure to select the best HTRX model: <code>do_cv</code>.
</p>
<p>Step 1: select candidate models using AIC, BIC or lasso;
</p>
<p>Step 2: select the best model using 10-fold cross-validation.
</p>
<p>There is also an option to directly perform 10-fold cross-validation: <code>do_cv_direct</code>.
This method loses some accuracy and doesn't return the fixed features selected, but saves computational time.
</p>
<p>Longer haplotypes are important for discovering interactions.
However, too many haplotypes make original HTRX unrealistic for regions with large numbers of SNPs.
We proposed "cumulative HTRX" that enables HTRX to run on longer haplotypes: <code>do_cumulative_htrx</code>.
</p>
<p>The code for HTRX is hosted at <a href="https://github.com/YaolingYang/HTRX">https://github.com/YaolingYang/HTRX</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Yaoling Yang <a href="mailto:yaoling.yang@bristol.ac.uk">yaoling.yang@bristol.ac.uk</a> (<a href="https://orcid.org/0000-0003-4905-8097">ORCID</a>)
</p>
<p>Authors:
</p>

<ul><li>
<p> Daniel Lawson <a href="mailto:Dan.Lawson@bristol.ac.uk">Dan.Lawson@bristol.ac.uk</a> (<a href="https://orcid.org/0000-0002-5311-6213">ORCID</a>)
</p>
</li></ul>
<h3>References</h3>

<p>Yang Y, Lawson DJ. HTRX: an R package for learning non-contiguous haplotypes associated with a phenotype. Bioinformatics Advances 3(1) (2023): vbad038.
</p>
<p>Barrie, W., Yang, Y., Irving-Pease, E.K. et al. Elevated genetic risk for multiple sclerosis emerged in steppe pastoralist populations. Nature 625, 321–328 (2024).
</p>
<p>Eforn, B. "Bootstrap methods: another look at the jackknife." The Annals of Statistics 7 (1979): 1-26.
</p>
<p>Schwarz, Gideon. "Estimating the dimension of a model." The annals of statistics (1978): 461-464.
</p>
<p>McFadden, Daniel. "Conditional logit analysis of qualitative choice behavior." (1973).
</p>
<p>Akaike, Hirotugu. "A new look at the statistical model identification." IEEE transactions on automatic control 19.6 (1974): 716-723.
</p>
<p>Tibshirani, Robert. "Regression shrinkage and selection via the lasso." Journal of the Royal Statistical Society: Series B (Methodological) 58.1 (1996): 267-288.
</p>


</div>