<div class="container">

<table style="width: 100%;"><tr>
<td>summary.HTSCluster</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Summarize results from clustering using a Poisson mixture model</h2>

<h3>Description</h3>

<p>A function to summarize the clustering results obtained from a Poisson mixture model.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'HTSCluster'
summary(object, ...)
## S3 method for class 'HTSClusterWrapper'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p> An object of class <code>"HTSCluster"</code> or <code>"HTSClusterWrapper"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The summary function for an object of class <code>"HTSCluster"</code> provides the following summary of results:
</p>
<p>1) Number of clusters and model selection criterion used, if applicable.
</p>
<p>2) Number of observations across all clusters with a maximum conditional probability
greater than 90
model.
</p>
<p>3) Number of observations per cluster with a maximum conditional probability
greater than 90
selected model.
</p>
<p>4) <code class="reqn">\boldsymbol{\lambda}</code> values for the selected model.
</p>
<p>5) <code class="reqn">\boldsymbol{\pi}</code> values for the selected model.
</p>
<p>The summary function for an object of class <code>"HTSClusterWrapper"</code> provides the number of clusters selected for
the BIC, ICL, DDSE, and Djump model selection approaches.
</p>


<h3>Author(s)</h3>

<p>Andrea Rau</p>


<h3>References</h3>

<p>Rau, A., Maugis-Rabusseau, C., Martin-Magniette, M.-L., Celeux G. (2015). Co-expression analysis of high-throughput transcriptome sequencing data with Poisson mixture models. Bioinformatics, 31(9):1420-1427.
</p>
<p>Rau, A., Celeux, G., Martin-Magniette, M.-L., Maugis-Rabusseau, C. (2011). Clustering high-throughput sequencing data with Poisson mixture models. Inria Research Report 7786. Available at <a href="https://inria.hal.science/inria-00638082">https://inria.hal.science/inria-00638082</a>.
</p>


<h3>See Also</h3>

<p><code>PoisMixClus</code>, <code>PoisMixClusWrapper</code></p>


<h3>Examples</h3>

<pre><code class="language-R">
set.seed(12345)

## Simulate data as shown in Rau et al. (2011)
## Library size setting "A", high cluster separation
## n = 2000 observations
simulate &lt;- PoisMixSim(n = 200, libsize = "A", separation = "high")
y &lt;- simulate$y
conds &lt;- simulate$conditions

## Run the PMM-II model for g = 3
## "TC" library size estimate, EM algorithm
run &lt;- PoisMixClus(y, g = 3, 
    norm = "TC", conds = conds, init.type = "small-em")

## Summary of results:
summary(run)


</code></pre>


</div>